{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "import yaml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"/home/dzeiberg/mave_preprocessing/datasets.yaml\") as f:\n",
    "    dataset_config = yaml.safe_load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Adamovich_BRCA1_Cisplatin': {'curation_name': 'Adamovich et al BRCA1 (Multiplex HDR assay)',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/publications/Adamovich_BRCA1/mmc2.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'sheet_name': 'Supp. Table S3',\n",
       "  'score_columns': ['FS1_BRCA1_siRNA',\n",
       "   'FS2_BRCA1_siRNA',\n",
       "   'FS3_BRCA1_siRNA',\n",
       "   'FS4_BRCA1_siRNA'],\n",
       "  'amino_acid_substitution_column': 'variantID',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P38398'},\n",
       " 'Adamovich_BRCA1_HDR': {'curation_name': 'Adamovich et al BRCA1 (Cisplatin resistance assay)',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/publications/Adamovich_BRCA1/mmc2.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'sheet_name': 'Supp. Table S2',\n",
       "  'score_columns': ['FS1_BRCA1_siRNA',\n",
       "   'FS2_BRCA1_siRNA',\n",
       "   'FS3_BRCA1_siRNA',\n",
       "   'FS4_BRCA1_siRNA'],\n",
       "  'amino_acid_substitution_column': 'variantID',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P38398'},\n",
       " 'Findlay_BRCA1_SGE': {'curation_name': 'Findlay et al BRCA1 SGE',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Findlay_BRCA1_SGE/raw/urn_mavedb_00000097-0-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro_with_transcript',\n",
       "  'nucleotide_substitution_column': 'hgvs_splice',\n",
       "  'nucleotide_substitution_format': 'hgvs_nuc_with_transcript',\n",
       "  'score_columns': ['score_rep1', 'score_rep2'],\n",
       "  'pipeline': 'B',\n",
       "  'uniprot_acc': 'P38398'},\n",
       " 'Hu_BRCA2_HDR': {'curation_name': 'BRCA2 Hu et al. 2024',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Hu_BRCA2_HDR/raw/1-s2.0-S000292972400034X-mmc2.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'header_row': 1,\n",
       "  'amino_acid_substitution_column': 'protein change',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'nucleotide_substitution_column': 'coding nucleotide change',\n",
       "  'nucleotide_substitution_format': 'hgvs_nuc',\n",
       "  'score_columns': ['HDR score'],\n",
       "  'pipeline': 'B',\n",
       "  'uniprot_acc': 'P51587'},\n",
       " 'Jia_MSH2_SSM': {'curation_name': 'Jia et al MSH2 SSM',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Jia_MSH2_SSM/raw/urn_mavedb_00000050-a-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'score_columns': ['score'],\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P43246',\n",
       "  'inverted': True},\n",
       " 'Kato_TP53': {'curation_name': 'Kato TP53',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Kato_TP53_AIP1nWT/raw/FunctionIshiokaDownload_r20.csv',\n",
       "  'extension': 'csv',\n",
       "  'note': 'median of 8 assays',\n",
       "  'pipeline': 'A',\n",
       "  'amino_acid_substitution_column': 'AAchange',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'score_columns': ['WAF1nWT',\n",
       "   'MDM2nWT',\n",
       "   'BAXnWT',\n",
       "   'h1433snWT',\n",
       "   'AIP1nWT',\n",
       "   'GADD45nWT',\n",
       "   'NOXAnWT',\n",
       "   'P53R2nWT'],\n",
       "  'score_agg_method': 'median',\n",
       "  'score_threshold_source': 'Fortuno TP53 VCEP meta (Kato)',\n",
       "  'postprocess_scores': ['normalize_B/LB'],\n",
       "  'author_transcript': 'NM_000546',\n",
       "  'uniprot_acc': 'P04637'},\n",
       " 'Giacomelli_1': {'curation_name': 'Giacomelli 1 (p53 WT Nutlin-3)',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Giacomelli_p53_wtNutlin3/raw/urn_mavedb_00000068-a-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'score_columns': ['score'],\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P04637',\n",
       "  'author_transcript': 'NM_000546',\n",
       "  'note': 'using transcript from Fayer et al. 2021 as I cannot find the transcript used in the original publication',\n",
       "  'inverted': True},\n",
       " 'Giacomelli_2': {'curation_name': 'Giacomelli 2 (p53 null Nutlin-3)',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Giacomelli_p53_nullNutlin3/raw/urn_mavedb_00000068-b-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'score_columns': ['score'],\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P04637',\n",
       "  'author_transcript': 'NM_000546',\n",
       "  'note': 'using transcript from Fayer et al. 2021 as I cannot find the transcript used in the original publication',\n",
       "  'inverted': True},\n",
       " 'Giacomelli_3': {'curation_name': 'Giacomelli 3 (p53 null etoposide)',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Giacomelli_p53_nullEtoposide/raw/urn_mavedb_00000068-c-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'score_columns': ['score'],\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P04637',\n",
       "  'author_transcript': 'NM_000546',\n",
       "  'note': 'using transcript from Fayer et al. 2021 as I cannot find the transcript used in the original publication'},\n",
       " 'Giacomelli_4': {'curation_name': 'Giacomelli 4 (combined phenotype score)',\n",
       "  'filepaths': ['/data/dzeiberg/mave_calibration/data/Giacomelli_p53_wtNutlin3/raw/urn_mavedb_00000068-a-1_scores.csv',\n",
       "   '/data/dzeiberg/mave_calibration/data/Giacomelli_p53_nullNutlin3/raw/urn_mavedb_00000068-b-1_scores.csv',\n",
       "   '/data/dzeiberg/mave_calibration/data/Giacomelli_p53_nullEtoposide/raw/urn_mavedb_00000068-c-1_scores.csv'],\n",
       "  'extensions': ['csv', 'csv', 'csv'],\n",
       "  'note': 'will need to go in, please calculate Z-score as: (Z-score(p53WT + nutlin) + Z-score(p53NULL + nutlin) âˆ’ Z-score(p53NULL + etoposide))/3',\n",
       "  'uniprot_acc': 'P04637'},\n",
       " 'Mighell_PTEN': {'curation_name': 'Mighell et al PTEN (phosphatase)',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/publications/Mighell_PTEN/mmc2.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'sheet_name': 'Table S2',\n",
       "  'header_row': 1,\n",
       "  'amino_acid_substitution_column': 'Variant (one letter)',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'score_columns': ['A1_score',\n",
       "   'A2_score',\n",
       "   'A3_score',\n",
       "   'B1_score',\n",
       "   'B2_score',\n",
       "   'B3_score'],\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P60484'},\n",
       " 'Matreyek_PTEN_VampSeq': {'curation_name': 'Matreyek et al PTEN (VAMPseq)',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Matreyek_PTEN_VampSeq/raw/urn_mavedb_00000013-a-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'score_columns': ['score1',\n",
       "   'score2',\n",
       "   'score3',\n",
       "   'score4',\n",
       "   'score5',\n",
       "   'score6',\n",
       "   'score7',\n",
       "   'score8'],\n",
       "  'author_transcript': 'NM_000314.8',\n",
       "  'note': 'using MANE-select transcript as none listed on curation',\n",
       "  'pipeline': 'A',\n",
       "  'uniprot_acc': 'P60484'},\n",
       " 'Glazer_SCN5A': {'curation_name': 'Glazer et al SCN5A Jan 2020',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/publications/Glazer_SCN5A/002786_supplemental_file s1.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'score_column': 'dms',\n",
       "  'classification_column': 'class',\n",
       "  'amino_acid_substitution_column': 'mutation',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'score_columns': ['dms'],\n",
       "  'pipeline': 'A',\n",
       "  'author_transcript': 'NM_198056',\n",
       "  'uniprot_acc': 'Q14524',\n",
       "  'postprocess_scores': ['normalize_synonymous']},\n",
       " 'Buckley_VHL_SGE': {'curation_name': 'Buckley et al VHL SGE',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/publications/Buckley_VHL/41588_2024_1800_MOESM4_ESM.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'classification_column': 'function_class',\n",
       "  'score_column': 'function_score_final',\n",
       "  'sheet_name': 'ST1',\n",
       "  'header_row': 2,\n",
       "  'amino_acid_substitution_column': 'pHGVS',\n",
       "  'amino_acid_substitution_format': 'p.single_letter',\n",
       "  'nucleotide_substitution_column': 'cHGVS',\n",
       "  'nucleotide_substitution_format': 'hgvs_nuc',\n",
       "  'pos_col': 'hg38_pos',\n",
       "  'ref_col': 'ref',\n",
       "  'alt_col': 'alt',\n",
       "  'CHROM': 3,\n",
       "  'score_columns': ['function_score_final'],\n",
       "  'pipeline': 'C',\n",
       "  'uniprot_acc': 'P40337',\n",
       "  'author_transcript': 'ENST00000256474.3'},\n",
       " 'BAP1_SGE': {'curation_name': 'BAP1 SGE',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Waters_BAP1_SGE/extended_data_1_sge_bap1_dataset.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'amino_acid_substitution_column': 'HGVSp',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro_with_transcript',\n",
       "  'sheet_name': 'extended_data_1',\n",
       "  'header_row': 2,\n",
       "  'nucleotide_substitution_column': 'HGVSc',\n",
       "  'nucleotide_substitution_format': 'hgvs_nuc_with_transcript',\n",
       "  'classification_column': 'functional_classification',\n",
       "  'score_columns': ['functional_score'],\n",
       "  'CHROM': 3,\n",
       "  'pos_col': 'pos',\n",
       "  'ref_col': 'ref',\n",
       "  'alt_col': 'alt',\n",
       "  'pipeline': 'C',\n",
       "  'uniprot_acc': 'Q92560'},\n",
       " 'Starita_BRCA1_HDR': {'curation_name': 'Starita et al BRCA1 HDR',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Starita_BRCA1_HDR/raw/urn_mavedb_00000081-a-2_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'uniprot_acc': 'P38398',\n",
       "  'author_transcript': 'NM_007294',\n",
       "  'note': 'using MANE-select transcript as none listed on curation',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'score_columns': ['brca1_sort1_score',\n",
       "   'brca1_sort2_score',\n",
       "   'brca1_sort3_score',\n",
       "   'brca1_sort4_score'],\n",
       "  'pipeline': 'A'},\n",
       " 'Boettcher_TP53': {'curation_name': 'Boetcher TP53',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/publications/Boettcher_TP53/TP53-MITEseq_collapsed_aa_fromIntendedCodons_inFraction_BoettcherSteffen.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'uniprot_acc': 'P04637',\n",
       "  'header_row': 0,\n",
       "  'amino_acid_substitution_column': 'mut',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'score_columns': ['SCORE'],\n",
       "  'pipeline': 'A'},\n",
       " 'Erwood_BRCA2_HEK293T': {'curation_name': 'Erwood et al. BRCA2',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Erwood_BRCA2_HEK293T/raw/41587_2021_1201_MOESM3_ESM.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'sheet_name': 'Supplementary Table 3',\n",
       "  'header_row': 1,\n",
       "  'uniprot_acc': 'P51587',\n",
       "  'amino_acid_substitution_column': 'Protein Annotation',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'classification_column': 'SPE Classification',\n",
       "  'score_columns': ['Function Score'],\n",
       "  'author_transcript': 'NM_000059',\n",
       "  'nucleotide_substitution_format': 'Erwood_custom',\n",
       "  'nucleotide_substitution_column': 'N/A',\n",
       "  'CHROM': 13,\n",
       "  'pos_col': 'end',\n",
       "  'ref_col': 'reference_base',\n",
       "  'alt_col': 'alternate_base',\n",
       "  'pipeline': 'C'},\n",
       " 'Erwood_NPC1_HEK293T': {'curation_name': 'Erwood et all NPC1 HEK239T',\n",
       "  'filepath': '/data/dzeiberg/mave_calibration/data/Erwood_NPC1_HEK293T/raw/41587_2021_1201_MOESM3_ESM.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'sheetname': 'Supplementary Table 1',\n",
       "  'header_row': 1,\n",
       "  'uniprot_acc': 'O15118',\n",
       "  'amino_acid_substitution_column': 'Protein Annotation',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'classification_column': 'SPE Classification',\n",
       "  'score_columns': ['Function Score'],\n",
       "  'author_transcript': 'NM_000271',\n",
       "  'nucleotide_substitution_format': 'Erwood_custom',\n",
       "  'nucleotide_substitution_column': 'N/A',\n",
       "  'CHROM': 18,\n",
       "  'pos_col': 'end',\n",
       "  'ref_col': 'reference_base',\n",
       "  'alt_col': 'alternate_base',\n",
       "  'pipeline': 'C'},\n",
       " 'Erwood_NPC1_RPE1': {'filepath': '/data/dzeiberg/mave_calibration/data/Erwood_NPC1_RPE1/raw/41587_2021_1201_MOESM3_ESM.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'sheetname': 'Supplementary Table 2',\n",
       "  'header_row': 1,\n",
       "  'curation_name': 'Erwood et all NPC1 RPE1',\n",
       "  'uniprot_acc': 'O15118',\n",
       "  'amino_acid_substitution_column': 'Protein Annotation',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'classification_column': 'SPE Classification',\n",
       "  'score_columns': ['Function Score'],\n",
       "  'author_transcript': 'NM_000271',\n",
       "  'nucleotide_substitution_format': 'Erwood_custom',\n",
       "  'nucleotide_substitution_column': 'N/A',\n",
       "  'CHROM': 18,\n",
       "  'pos_col': 'end',\n",
       "  'ref_col': 'reference_base',\n",
       "  'alt_col': 'alternate_base',\n",
       "  'pipeline': 'C'},\n",
       " 'Weile_CALM': {'filepath': '/data/dzeiberg/mave_calibration/publications/Weile/CALM123/urn_mavedb_00000001-c-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'curation_name': 'CALM1/2/3 Yeast Complemtation',\n",
       "  'uniprot_accessions': ['P0DP23', 'P0DP24', 'P0DP25'],\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'nucleotide_substitution_column': 'hgvs_nt',\n",
       "  'nucleotide_substitution_format': 'hgvs_nuc',\n",
       "  'pipeline': 'A',\n",
       "  'author_transcript': 'NM_001363670',\n",
       "  'transcript_offset': 833,\n",
       "  'expecting_duplicates_in_scoreset': True,\n",
       "  'score_columns': ['score'],\n",
       "  'HGNC_IDs': ['HGNC:1442', 'HGNC:1445', 'HGNC:1449']},\n",
       " 'Weile_TPK1': {'filepath': '/data/dzeiberg/mave_calibration/publications/Weile/TPK/urn_mavedb_00000001-d-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'curation_name': 'TPK1 Yeast Complemtation',\n",
       "  'uniprot_acc': 'Q9H3S4',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'nucleotide_substitution_column': 'hgvs_nt',\n",
       "  'nucleotide_substitution_format': 'hgvs_nuc',\n",
       "  'pipeline': 'B',\n",
       "  'author_transcript': 'NM_022445.3',\n",
       "  'transcript_offset': 104,\n",
       "  'score_columns': ['score']},\n",
       " 'Lo_OTC': {'filepath': '/data/dzeiberg/mave_calibration/publications/Lo_OTC/urn_mavedb_00000112-a-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'uniprot_acc': 'P00480',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'score_columns': ['score'],\n",
       "  'pipeline': 'A'},\n",
       " 'RAD51C': {'filepath': '/data/dzeiberg/mave_calibration/publications/RAD51C/mmc1.xlsx',\n",
       "  'extension': 'xlsx',\n",
       "  'sheet_name': 'SGE functional classifications',\n",
       "  'header_row': 2,\n",
       "  'curation_name': 'Olvera-LeÃ³n et al RAD51C 2024',\n",
       "  'uniprot_acc': 'O43502',\n",
       "  'amino_acid_substitution_column': 'HGVSp',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro_with_transcript',\n",
       "  'nucleotide_substitution_column': 'HGVSc',\n",
       "  'nucleotide_substitution_format': 'hgvs_nuc_with_transcript',\n",
       "  'classification_column': 'functional_classification',\n",
       "  'score_columns': ['z_score_D4_D14'],\n",
       "  'pipeline': 'C',\n",
       "  'CHROM': 17,\n",
       "  'pos_col': 'pos',\n",
       "  'ref_col': 'ref',\n",
       "  'alt_col': 'alt',\n",
       "  'expecting_duplicates_in_scoreset': True},\n",
       " 'Ollodart_MSH2': {'filepath': '/data/dzeiberg/mave_calibration/publications/Ollodart_MSH2/filtered.csv',\n",
       "  'extension': 'csv',\n",
       "  'curation_name': 'Ollodart et. al. MSH2 2021 - barcoded',\n",
       "  'uniprot_acc': 'P43246',\n",
       "  'amino_acid_substitution_column': 'Human Msh2 Genotypeb',\n",
       "  'amino_acid_substitution_format': 'single_letter',\n",
       "  'score_columns': ['Scoreg'],\n",
       "  'pipeline': 'A',\n",
       "  'expecting_duplicates_in_scoreset': True},\n",
       " 'vanLoggerenberg_HMBS': {'filepath': '/data/dzeiberg/mave_calibration/publications/vanLoggerenberg_HMBS/urn_mavedb_00000108-a-3_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'curation_name': 'HMBS Loggerenberg et al. - combined isoforms',\n",
       "  'uniprot_acc': 'P08397',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'score_columns': ['score'],\n",
       "  'pipeline': 'A'},\n",
       " 'PRKN_VampSeq': {'filepath': '/data/dzeiberg/mave_calibration/publications/PRKN_VampSeq/urn_mavedb_00000114-a-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'curation_name': 'Clausen et al Parkin 2024 VampSeq',\n",
       "  'uniprot_acc': 'O60260',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'score_columns': ['score'],\n",
       "  'pipeline': 'A'},\n",
       " 'Gray_AB42': {'filepath': '/data/dzeiberg/mave_calibration/publications/Gray_AB42/urn_mavedb_00000058-a-1_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'curation_name': 'Gray et al amyloid beta 42 2019',\n",
       "  'protein_offset': 670,\n",
       "  'uniprot_acc': 'P05067',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'score_columns': ['score'],\n",
       "  'pipeline': 'A'},\n",
       " 'Seuma_AB42': {'filepath': '/data/dzeiberg/mave_calibration/publications/Seuma_AB42/urn_mavedb_00000113-a-3_scores.csv',\n",
       "  'extension': 'csv',\n",
       "  'curation_name': 'Seuma et al amyloid beta 42 2021',\n",
       "  'uniprot_acc': 'P05067',\n",
       "  'amino_acid_substitution_column': 'hgvs_pro',\n",
       "  'amino_acid_substitution_format': 'hgvs_pro',\n",
       "  'score_columns': ['score'],\n",
       "  'pipeline': 'A'}}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset_config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset_path = Path(\"/data/dzeiberg/mave_calibration/processed_datasets/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "datasets = {file.stem.split(\"_pipeline\")[0] : pd.read_json(file) for file in dataset_path.glob(\"*.json\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts = {dataset_config[name]['curation_name'] if name in dataset_config and \"curation_name\" in dataset_config[name] else name : {\"P/LP\" : df.labels.apply(lambda li: \"P/LP\" in li).sum(),\n",
    "                  \"B/LB\" : df.labels.apply(lambda li: \"B/LB\" in li).sum(),\n",
    "                  \"gnomAD\": df.labels.apply(lambda li: \"gnomAD\" in li).sum(),\n",
    "                  \"synonymous\": df.labels.apply(lambda li: \"synonymous\" in li).sum(),\n",
    "                  \"VUS\" : df.labels.apply(lambda li: \"VUS\" in li).sum(),} for name, df in datasets.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(counts).T.to_latex(\"datasets.tex\",)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mave",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
