{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1344924/312130268.py:5: TqdmExperimentalWarning: Using `tqdm.autonotebook.tqdm` in notebook mode. Use `tqdm.tqdm` instead to force console mode (e.g. in jupyter console)\n",
      "  from tqdm.autonotebook import tqdm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import yaml\n",
    "import re\n",
    "from tqdm.autonotebook import tqdm\n",
    "from pathlib import Path\n",
    "pd.set_option('display.max_columns', 500)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "evidenceDF = pd.read_csv(\"/data/dzeiberg/mave_calibration/clingenEvidenceRepo/erepo.tabbed.txt\",delimiter='\\t')\n",
    "evidenceDF = evidenceDF.assign(GeneSymbol=evidenceDF.loc[:,'HGVS Expressions'].apply(lambda x: re.search(r\"\\(([^)]+)\\)\", x).group(1) if re.search(r\"\\(([^)]+)\\)\", x) else None))\n",
    "evidenceDF = evidenceDF.assign(metEvidence=evidenceDF.loc[:,'Applied Evidence Codes (Met)'].astype(str).str.split(',').apply(lambda li: [str(x).strip() for x in li]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"datasets.yaml\") as f:\n",
    "    datasets = yaml.safe_load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = set([ds['gene'].upper() for ds in datasets.values() if 'gene' in ds])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "evidenceDF = evidenceDF[evidenceDF.GeneSymbol.str.upper().isin(genes)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "datadir = Path(\"/data/dzeiberg/mave_calibration/processed_datasets\")\n",
    "scoresets = {}\n",
    "for dataset_name, dsInfo in datasets.items():\n",
    "    if \"pipeline\" in dsInfo:\n",
    "        pipeline = dsInfo[\"pipeline\"]\n",
    "        scoreset_file = datadir / f\"{dataset_name}_pipeline_{pipeline}.json\"\n",
    "        if scoreset_file.exists():\n",
    "            scoresets[dataset_name] = pd.read_json(scoreset_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "25"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(scoresets)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Variation</th>\n",
       "      <th>ClinVar Variation Id</th>\n",
       "      <th>Allele Registry Id</th>\n",
       "      <th>HGVS Expressions</th>\n",
       "      <th>HGNC Gene Symbol</th>\n",
       "      <th>Disease</th>\n",
       "      <th>Mondo Id</th>\n",
       "      <th>Mode of Inheritance</th>\n",
       "      <th>Assertion</th>\n",
       "      <th>Applied Evidence Codes (Met)</th>\n",
       "      <th>Applied Evidence Codes (Not Met)</th>\n",
       "      <th>Summary of interpretation</th>\n",
       "      <th>PubMed Articles</th>\n",
       "      <th>Expert Panel</th>\n",
       "      <th>Guideline</th>\n",
       "      <th>Approval Date</th>\n",
       "      <th>Published Date</th>\n",
       "      <th>Retracted</th>\n",
       "      <th>Evidence Repo Link</th>\n",
       "      <th>Uuid</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>metEvidence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>NM_000314.6(PTEN):c.737C&gt;T (p.Pro246Leu)</td>\n",
       "      <td>142269</td>\n",
       "      <td>CA000559</td>\n",
       "      <td>NM_000314.6:c.737C&gt;T, NM_000314.5:c.737C&gt;T, NM...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PM2, PS2_Very Strong, PS4, PP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.737C&gt;T (p.P246L) meets criteria to be c...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-10-18</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>835a14e6-4a41-47df-a1ff-271c872adf39</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PM2, PS2_Very Strong, PS4, PP2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>NM_000314.6(PTEN):c.-1170C&gt;T</td>\n",
       "      <td>127662</td>\n",
       "      <td>CA151481</td>\n",
       "      <td>NM_000314.6:c.-1170C&gt;T, NM_001126049.1:c.-812G...</td>\n",
       "      <td>KLLN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>BP5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.-1170C&gt;T (NC_000010.10: g.89623056C&gt;T) ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-11-08</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>8c50af60-fe53-4c9b-b60a-3735d21c8645</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[BP5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>NM_000314.6(PTEN):c.209+3A&gt;T</td>\n",
       "      <td>185989</td>\n",
       "      <td>CA000135</td>\n",
       "      <td>NM_000314.6:c.209+3A&gt;T, NM_000314.5:c.209+3A&gt;T...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>PP3</td>\n",
       "      <td>PM2, PM6, PVS1, BA1, BP4, BP2, PS3, PS1, PS2, ...</td>\n",
       "      <td>PTEN c.209+3A&gt;T (IVS3+3A&gt;T) is currently class...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020-03-23</td>\n",
       "      <td>2020-03-30</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>049d3e03-b5ad-4a47-92e9-ee077179049a</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PP3]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>NM_000314.6(PTEN):c.235G&gt;A (p.Ala79Thr)</td>\n",
       "      <td>41682</td>\n",
       "      <td>CA000358</td>\n",
       "      <td>NM_000314.6:c.235G&gt;A, NM_000314.5:c.235G&gt;A, NM...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>BS2_Supporting, BS1, PP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.235G&gt;A (p.A79T) meets criteria to be cl...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-04-06</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>0340272b-ab4c-4672-94b8-40e0525d8eda</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[BS2_Supporting, BS1, PP2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>NM_000314.6(PTEN):c.304_306dupAAA (p.Lys102_Pr...</td>\n",
       "      <td>142681</td>\n",
       "      <td>CA169101</td>\n",
       "      <td>NM_000314.6:c.304_306dupAAA, NM_000314.5:c.304...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>PM2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.304_306dupAAA (p.K102_P103insK) is curr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-11-08</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>742e08c9-7bda-4006-95f1-22f34ec0d9e1</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PM2]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           #Variation ClinVar Variation Id  \\\n",
       "82           NM_000314.6(PTEN):c.737C>T (p.Pro246Leu)               142269   \n",
       "83                       NM_000314.6(PTEN):c.-1170C>T               127662   \n",
       "84                       NM_000314.6(PTEN):c.209+3A>T               185989   \n",
       "85            NM_000314.6(PTEN):c.235G>A (p.Ala79Thr)                41682   \n",
       "86  NM_000314.6(PTEN):c.304_306dupAAA (p.Lys102_Pr...               142681   \n",
       "\n",
       "   Allele Registry Id                                   HGVS Expressions  \\\n",
       "82           CA000559  NM_000314.6:c.737C>T, NM_000314.5:c.737C>T, NM...   \n",
       "83           CA151481  NM_000314.6:c.-1170C>T, NM_001126049.1:c.-812G...   \n",
       "84           CA000135  NM_000314.6:c.209+3A>T, NM_000314.5:c.209+3A>T...   \n",
       "85           CA000358  NM_000314.6:c.235G>A, NM_000314.5:c.235G>A, NM...   \n",
       "86           CA169101  NM_000314.6:c.304_306dupAAA, NM_000314.5:c.304...   \n",
       "\n",
       "   HGNC Gene Symbol                        Disease       Mondo Id  \\\n",
       "82             PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "83             KLLN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "84             PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "85             PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "86             PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "\n",
       "               Mode of Inheritance               Assertion  \\\n",
       "82  Autosomal dominant inheritance              Pathogenic   \n",
       "83  Autosomal dominant inheritance  Uncertain Significance   \n",
       "84  Autosomal dominant inheritance  Uncertain Significance   \n",
       "85  Autosomal dominant inheritance           Likely Benign   \n",
       "86  Autosomal dominant inheritance  Uncertain Significance   \n",
       "\n",
       "      Applied Evidence Codes (Met)  \\\n",
       "82  PM2, PS2_Very Strong, PS4, PP2   \n",
       "83                             BP5   \n",
       "84                             PP3   \n",
       "85        BS2_Supporting, BS1, PP2   \n",
       "86                             PM2   \n",
       "\n",
       "                     Applied Evidence Codes (Not Met)  \\\n",
       "82                                                NaN   \n",
       "83                                                NaN   \n",
       "84  PM2, PM6, PVS1, BA1, BP4, BP2, PS3, PS1, PS2, ...   \n",
       "85                                                NaN   \n",
       "86                                                NaN   \n",
       "\n",
       "                            Summary of interpretation PubMed Articles  \\\n",
       "82  PTEN c.737C>T (p.P246L) meets criteria to be c...             NaN   \n",
       "83  PTEN c.-1170C>T (NC_000010.10: g.89623056C>T) ...             NaN   \n",
       "84  PTEN c.209+3A>T (IVS3+3A>T) is currently class...             NaN   \n",
       "85  PTEN c.235G>A (p.A79T) meets criteria to be cl...             NaN   \n",
       "86  PTEN c.304_306dupAAA (p.K102_P103insK) is curr...             NaN   \n",
       "\n",
       "   Expert Panel Guideline Approval Date Published Date  Retracted  \\\n",
       "82    PTEN VCEP       NaN    2017-10-18     2018-12-10      False   \n",
       "83    PTEN VCEP       NaN    2017-11-08     2018-12-10      False   \n",
       "84    PTEN VCEP       NaN    2020-03-23     2020-03-30      False   \n",
       "85    PTEN VCEP       NaN    2018-04-06     2018-12-10      False   \n",
       "86    PTEN VCEP       NaN    2017-11-08     2018-12-10      False   \n",
       "\n",
       "                                   Evidence Repo Link  \\\n",
       "82  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "83  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "84  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "85  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "86  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "\n",
       "                                    Uuid GeneSymbol  \\\n",
       "82  835a14e6-4a41-47df-a1ff-271c872adf39       PTEN   \n",
       "83  8c50af60-fe53-4c9b-b60a-3735d21c8645       PTEN   \n",
       "84  049d3e03-b5ad-4a47-92e9-ee077179049a       PTEN   \n",
       "85  0340272b-ab4c-4672-94b8-40e0525d8eda       PTEN   \n",
       "86  742e08c9-7bda-4006-95f1-22f34ec0d9e1       PTEN   \n",
       "\n",
       "                         metEvidence  \n",
       "82  [PM2, PS2_Very Strong, PS4, PP2]  \n",
       "83                             [BP5]  \n",
       "84                             [PP3]  \n",
       "85        [BS2_Supporting, BS1, PP2]  \n",
       "86                             [PM2]  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "evidenceDF.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(381, 22)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "evidenceDF.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "evidenceDF = evidenceDF.assign(Expressions=evidenceDF.loc[:,'HGVS Expressions'].str.split(', ').apply(tuple))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_007294.4:c.1233T>G\n",
      "NM_007294.4:c.135-1G>T\n",
      "p.Asp411Glu\n",
      "p.Gly12Ser\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "def extract_transcript_and_variant(coding_string):\n",
    "    # Regex pattern to match the transcript, allowing for optional parentheses (gene name) and capturing the variant\n",
    "    pattern = r\"(NM_\\d+\\.\\d+)(?:\\([A-Za-z0-9_]+\\))?:?(c\\.\\d+[-]?\\d*[A-Z]>[A-Z])\"\n",
    "    match = re.search(pattern, coding_string)\n",
    "    if match:\n",
    "        transcript = match.group(1)\n",
    "        variant = match.group(2)\n",
    "        return f\"{transcript}:{variant}\"\n",
    "    return None\n",
    "\n",
    "def extract_hgvs_nuc(coding_string):\n",
    "    # Updated regex pattern to match both \"c.1233T>G\" and \"c.135-1G>T\" formats\n",
    "    pattern = r\"(c\\.\\d+[-]?\\d*[A-Z]>[A-Z])\"\n",
    "    match = re.search(pattern, coding_string)\n",
    "    if match:\n",
    "        return match.group(1)\n",
    "    return None\n",
    "\n",
    "# Example usage\n",
    "string1 = \"NM_007294.4(BRCA1):c.1233T>G (p.Asp411Glu)\"\n",
    "string2 = \"NM_007294.4:c.135-1G>T\"\n",
    "\n",
    "# Example usage\n",
    "string1 = \"NM_007294.4(BRCA1):c.1233T>G (p.Asp411Glu)\"\n",
    "string2 = \"NM_007294.4:c.135-1G>T\"\n",
    "result1 = extract_transcript_and_variant(string1)\n",
    "result2 = extract_transcript_and_variant(string2)\n",
    "\n",
    "print(result1)  # Output: NM_007294.4:c.1233T>G\n",
    "print(result2)  # Output: NM_007294.4:c.135-1G>T\n",
    "\n",
    "def extract_amino_acid_substitution(coding_string):\n",
    "    # Regex pattern to match the amino acid substitution in the form \"p.Asp411Glu\"\n",
    "    pattern = r\"(p\\.[A-Z][a-z]{2}\\d+[A-Z][a-z]{2})\"\n",
    "    match = re.search(pattern, coding_string)\n",
    "    if match:\n",
    "        return match.group(1)\n",
    "    return None\n",
    "\n",
    "# Example usage\n",
    "string1 = \"NM_007294.4(BRCA1):c.1233T>G (p.Asp411Glu)\"\n",
    "string2 = \"NM_007294.4:c.135-1G>T (p.Gly12Ser)\"\n",
    "result1 = extract_amino_acid_substitution(string1)\n",
    "result2 = extract_amino_acid_substitution(string2)\n",
    "\n",
    "print(result1)  # Output: p.Asp411Glu\n",
    "print(result2)  # Output: p.Gly12Ser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "evidenceDF = evidenceDF.assign(hgvs_nuc_full=evidenceDF.Expressions.apply(lambda li: tuple(list(set((extract_transcript_and_variant(x) for x in li)) - {None,}))),\n",
    "                               hgvs_nuc=evidenceDF.Expressions.apply(lambda li: tuple(list(set((extract_hgvs_nuc(x) for x in li)) - {None,}))),\n",
    "                               hgvs_pro=evidenceDF.Expressions.apply(lambda li: tuple(list(set((extract_amino_acid_substitution(x) for x in li)) - {None,}))))\n",
    "evidenceDF = evidenceDF.assign(hgvs_pro=evidenceDF.hgvs_pro.apply(lambda x: x[0] if len(x) else \"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "scoreset_with_evidence = {}\n",
    "for scoresetName, ss in scoresets.items():\n",
    "    ss= ss.dropna(subset=['author_transcript','hgvs_pro'])\n",
    "    ss = ss[ss.hgvs_pro.str.len() > 0]\n",
    "    if 'author_transcript' not in ss.columns:\n",
    "        ss = ss.assign(author_transcript=datasets[scoresetName]['author_transcript'])\n",
    "    ss = ss.assign(avg_score=ss.scores.apply(lambda x: sum(x)/len(x)))\n",
    "    sse = pd.merge(ss, evidenceDF[evidenceDF.GeneSymbol == datasets[scoresetName]['gene']], on=['hgvs_pro'],suffixes=('_mave', '_errepo'),how='inner')\n",
    "    scoreset_with_evidence[scoresetName] = sse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adamovich_BRCA1_Cisplatin: 2\n",
      "Adamovich_BRCA1_HDR: 2\n",
      "Findlay_BRCA1_SGE: 4\n",
      "Hu_BRCA2_HDR: 5\n",
      "Kato_TP53: 108\n",
      "Giacomelli_1: 108\n",
      "Giacomelli_2: 108\n",
      "Giacomelli_3: 108\n",
      "Mighell_PTEN: 114\n",
      "Matreyek_PTEN_VampSeq: 73\n",
      "Buckley_VHL_SGE: 22\n",
      "Starita_BRCA1_HDR: 1\n"
     ]
    }
   ],
   "source": [
    "for scoresetName, sse in scoreset_with_evidence.items():\n",
    "    if sse.shape[0]:\n",
    "        print(f\"{scoresetName}: {sse.shape[0]:,d}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1344924/137977562.py:1: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  df = pd.concat([scoreset_with_evidence[dataset_name].loc[:,['hgvs_pro','avg_score','#Variation', 'ClinVar Variation Id', 'Allele Registry Id',\n"
     ]
    }
   ],
   "source": [
    "df = pd.concat([scoreset_with_evidence[dataset_name].loc[:,['hgvs_pro','avg_score','#Variation', 'ClinVar Variation Id', 'Allele Registry Id',\n",
    "       'HGVS Expressions', 'HGNC Gene Symbol', 'Disease', 'Mondo Id',\n",
    "       'Mode of Inheritance', 'Assertion', 'Applied Evidence Codes (Met)',\n",
    "       'Applied Evidence Codes (Not Met)', 'Summary of interpretation',\n",
    "       'PubMed Articles', 'Expert Panel', 'Guideline', 'Approval Date',\n",
    "       'Published Date', 'Retracted', 'Evidence Repo Link', 'Uuid',\n",
    "       'GeneSymbol', 'metEvidence', 'Expressions', 'hgvs_pro']].assign(scoreset_source=dataset_name) for dataset_name in scoreset_with_evidence.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(253,)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[:,'#Variation'].unique().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Variation</th>\n",
       "      <th>ClinVar Variation Id</th>\n",
       "      <th>Allele Registry Id</th>\n",
       "      <th>HGVS Expressions</th>\n",
       "      <th>HGNC Gene Symbol</th>\n",
       "      <th>Disease</th>\n",
       "      <th>Mondo Id</th>\n",
       "      <th>Mode of Inheritance</th>\n",
       "      <th>Assertion</th>\n",
       "      <th>Applied Evidence Codes (Met)</th>\n",
       "      <th>Applied Evidence Codes (Not Met)</th>\n",
       "      <th>Summary of interpretation</th>\n",
       "      <th>PubMed Articles</th>\n",
       "      <th>Expert Panel</th>\n",
       "      <th>Guideline</th>\n",
       "      <th>Approval Date</th>\n",
       "      <th>Published Date</th>\n",
       "      <th>Retracted</th>\n",
       "      <th>Evidence Repo Link</th>\n",
       "      <th>Uuid</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>metEvidence</th>\n",
       "      <th>Expressions</th>\n",
       "      <th>hgvs_nuc_full</th>\n",
       "      <th>hgvs_nuc</th>\n",
       "      <th>hgvs_pro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>NM_000314.6(PTEN):c.737C&gt;T (p.Pro246Leu)</td>\n",
       "      <td>142269</td>\n",
       "      <td>CA000559</td>\n",
       "      <td>NM_000314.6:c.737C&gt;T, NM_000314.5:c.737C&gt;T, NM...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PM2, PS2_Very Strong, PS4, PP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.737C&gt;T (p.P246L) meets criteria to be c...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-10-18</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>835a14e6-4a41-47df-a1ff-271c872adf39</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PM2, PS2_Very Strong, PS4, PP2]</td>\n",
       "      <td>(NM_000314.6:c.737C&gt;T, NM_000314.5:c.737C&gt;T, N...</td>\n",
       "      <td>(NM_001304717.2:c.1256C&gt;T, NM_000314.5:c.737C&gt;...</td>\n",
       "      <td>(c.692C&gt;T, c.641C&gt;T, c.1256C&gt;T, c.737C&gt;T, c.14...</td>\n",
       "      <td>p.Pro246Leu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>NM_000314.6(PTEN):c.235G&gt;A (p.Ala79Thr)</td>\n",
       "      <td>41682</td>\n",
       "      <td>CA000358</td>\n",
       "      <td>NM_000314.6:c.235G&gt;A, NM_000314.5:c.235G&gt;A, NM...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>BS2_Supporting, BS1, PP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.235G&gt;A (p.A79T) meets criteria to be cl...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-04-06</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>0340272b-ab4c-4672-94b8-40e0525d8eda</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[BS2_Supporting, BS1, PP2]</td>\n",
       "      <td>(NM_000314.6:c.235G&gt;A, NM_000314.5:c.235G&gt;A, N...</td>\n",
       "      <td>(NM_001304717.5:c.754G&gt;A, NM_000314.7:c.235G&gt;A...</td>\n",
       "      <td>(c.235G&gt;A, c.133G&gt;A, c.190G&gt;A, c.754G&gt;A, c.139...</td>\n",
       "      <td>p.Ala79Thr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>NM_000314.6(PTEN):c.987_990delTAAA (p.Asn329Ly...</td>\n",
       "      <td>189441</td>\n",
       "      <td>CA000656</td>\n",
       "      <td>NM_000314.6:c.987_990delTAAA, NM_000314.5:c.98...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PM2, PVS1, PS4_Supporting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.987_990delTAAA (p.N329KfsX14) meets cri...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-10-18</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>e41ad782-4c9c-4ae6-ba02-ec2e11c967fc</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PM2, PVS1, PS4_Supporting]</td>\n",
       "      <td>(NM_000314.6:c.987_990delTAAA, NM_000314.5:c.9...</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>p.Asn329Lys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NM_000314.6(PTEN):c.892C&gt;T (p.Gln298Ter)</td>\n",
       "      <td>187657</td>\n",
       "      <td>CA000219</td>\n",
       "      <td>NM_000314.6:c.892C&gt;T, NC_000010.11:g.87960984C...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PM2, PVS1, PS4_Supporting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN c.892C&gt;T (p.Q298X) meets criteria to be c...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-10-18</td>\n",
       "      <td>2018-12-10</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>f890d81d-90ed-4411-a0fd-a89669dd616a</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PM2, PVS1, PS4_Supporting]</td>\n",
       "      <td>(NM_000314.6:c.892C&gt;T, NC_000010.11:g.87960984...</td>\n",
       "      <td>(NM_001304718.2:c.301C&gt;T, NM_001304718.1:c.301...</td>\n",
       "      <td>(c.796C&gt;T, c.1411C&gt;T, c.892C&gt;T, c.301C&gt;T, c.84...</td>\n",
       "      <td>p.Gln298Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>NM_000314.6(PTEN):c.964A&gt;T (p.Lys322Ter)</td>\n",
       "      <td>185213</td>\n",
       "      <td>CA000235</td>\n",
       "      <td>NM_000314.6:c.964A&gt;T, NC_000010.11:g.87961056A...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PVS1, PM2</td>\n",
       "      <td>PM6, BP2, BP4, BA1, PS4, PS2, PS1, PS3, BS4, B...</td>\n",
       "      <td>PTEN c.964A&gt;T (p.Lys322Ter) meets criteria to ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020-03-23</td>\n",
       "      <td>2020-03-26</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>bfdfbf81-6f26-43f8-aa4b-bbb17ebaf54a</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PVS1, PM2]</td>\n",
       "      <td>(NM_000314.6:c.964A&gt;T, NC_000010.11:g.87961056...</td>\n",
       "      <td>(NM_001304717.5:c.1483A&gt;T, NM_001304717.2:c.14...</td>\n",
       "      <td>(c.919A&gt;T, c.964A&gt;T, c.868A&gt;T, c.1483A&gt;T, c.37...</td>\n",
       "      <td>p.Lys322Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7226</th>\n",
       "      <td>NM_000551.4(VHL):c.257C&gt;T (p.Pro86Leu)</td>\n",
       "      <td>182977</td>\n",
       "      <td>CA020186</td>\n",
       "      <td>NM_000551.4:c.257C&gt;T, NC_000003.12:g.10142104C...</td>\n",
       "      <td>VHL</td>\n",
       "      <td>von Hippel-Lindau disease</td>\n",
       "      <td>MONDO:0008667</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PM2_Supporting, PS4, PP1_Strong</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The variant NM_000551.4(VHL):c.257C&gt;T (p.Pro86...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VHL VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>318690ae-2743-4195-8d9b-e2e84ff94494</td>\n",
       "      <td>VHL</td>\n",
       "      <td>[PM2_Supporting, PS4, PP1_Strong]</td>\n",
       "      <td>(NM_000551.4:c.257C&gt;T, NC_000003.12:g.10142104...</td>\n",
       "      <td>(NM_198156.3:c.257C&gt;T, NM_198156.2:c.257C&gt;T, N...</td>\n",
       "      <td>(c.257C&gt;T,)</td>\n",
       "      <td>p.Pro86Leu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7228</th>\n",
       "      <td>NM_000551.4(VHL):c.241C&gt;T (p.Pro81Ser)</td>\n",
       "      <td>2233</td>\n",
       "      <td>CA020148</td>\n",
       "      <td>NM_000551.4:c.241C&gt;T, NC_000003.12:g.10142088C...</td>\n",
       "      <td>VHL</td>\n",
       "      <td>von Hippel-Lindau disease</td>\n",
       "      <td>MONDO:0008667</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Benign</td>\n",
       "      <td>BA1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The variant NM_000551.4(VHL):c.241C&gt;T (p.Pro81...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VHL VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>416afbe6-3dd6-40d9-9e61-d3c54b017b5e</td>\n",
       "      <td>VHL</td>\n",
       "      <td>[BA1]</td>\n",
       "      <td>(NM_000551.4:c.241C&gt;T, NC_000003.12:g.10142088...</td>\n",
       "      <td>(NM_000551.3:c.241C&gt;T, NM_001354723.2:c.241C&gt;T...</td>\n",
       "      <td>(c.241C&gt;T,)</td>\n",
       "      <td>p.Pro81Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7231</th>\n",
       "      <td>NM_000551.4(VHL):c.154G&gt;A (p.Glu52Lys)</td>\n",
       "      <td>161402</td>\n",
       "      <td>CA020056</td>\n",
       "      <td>NM_000551.4:c.154G&gt;A, NC_000003.12:g.10142001G...</td>\n",
       "      <td>VHL</td>\n",
       "      <td>von Hippel-Lindau disease</td>\n",
       "      <td>MONDO:0008667</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Benign</td>\n",
       "      <td>BA1</td>\n",
       "      <td>PP3</td>\n",
       "      <td>The variant NM_000551.3(VHL):c.154G&gt;A (p.Glu52...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VHL VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>9f120c06-bb61-4115-9c9c-187ef481f769</td>\n",
       "      <td>VHL</td>\n",
       "      <td>[BA1]</td>\n",
       "      <td>(NM_000551.4:c.154G&gt;A, NC_000003.12:g.10142001...</td>\n",
       "      <td>(NM_198156.3:c.154G&gt;A, NM_000551.4:c.154G&gt;A, N...</td>\n",
       "      <td>(c.154G&gt;A,)</td>\n",
       "      <td>p.Glu52Lys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7418</th>\n",
       "      <td>NM_000551.4(VHL):c.263G&gt;A (p.Trp88Ter)</td>\n",
       "      <td>182978</td>\n",
       "      <td>CA020197</td>\n",
       "      <td>NM_000551.4:c.263G&gt;A, NC_000003.12:g.10142110G...</td>\n",
       "      <td>VHL</td>\n",
       "      <td>von Hippel-Lindau disease</td>\n",
       "      <td>MONDO:0008667</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PVS1, PM2_Supporting, PS4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The variant NM_000551.4(VHL):c.263G&gt;A (p.Trp88...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VHL VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>2024-07-17</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>eb3883a6-80f5-45f8-9c0f-a2928a4e24ca</td>\n",
       "      <td>VHL</td>\n",
       "      <td>[PVS1, PM2_Supporting, PS4]</td>\n",
       "      <td>(NM_000551.4:c.263G&gt;A, NC_000003.12:g.10142110...</td>\n",
       "      <td>(NM_198156.2:c.263G&gt;A, NM_001354723.1:c.263G&gt;A...</td>\n",
       "      <td>(c.263G&gt;A,)</td>\n",
       "      <td>p.Trp88Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7862</th>\n",
       "      <td>NM_000546.6(TP53):c.105G&gt;T (p.Leu35Phe)</td>\n",
       "      <td>12371</td>\n",
       "      <td>CA000025</td>\n",
       "      <td>NM_000546.6:c.105G&gt;T, NC_000017.11:g.7676264C&gt;...</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Li-Fraumeni syndrome</td>\n",
       "      <td>MONDO:0018875</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>PM2_Supporting, BP4</td>\n",
       "      <td>PM5, PM1, PP3, PP4, PP1, BS2, BA1, PS1, PS3, P...</td>\n",
       "      <td>The NM_000546.6: c.105G&gt;T variant in TP53 is a...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53 VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-09-06</td>\n",
       "      <td>2024-09-06</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>b73a437c-26a5-4491-8b92-edbe32570d9e</td>\n",
       "      <td>TP53</td>\n",
       "      <td>[PM2_Supporting, BP4]</td>\n",
       "      <td>(NM_000546.6:c.105G&gt;T, NC_000017.11:g.7676264C...</td>\n",
       "      <td>(NM_000546.6:c.105G&gt;T, NM_001126114.2:c.105G&gt;T...</td>\n",
       "      <td>(c.105G&gt;T,)</td>\n",
       "      <td>p.Leu35Phe</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             #Variation ClinVar Variation Id  \\\n",
       "82             NM_000314.6(PTEN):c.737C>T (p.Pro246Leu)               142269   \n",
       "85              NM_000314.6(PTEN):c.235G>A (p.Ala79Thr)                41682   \n",
       "94    NM_000314.6(PTEN):c.987_990delTAAA (p.Asn329Ly...               189441   \n",
       "95             NM_000314.6(PTEN):c.892C>T (p.Gln298Ter)               187657   \n",
       "96             NM_000314.6(PTEN):c.964A>T (p.Lys322Ter)               185213   \n",
       "...                                                 ...                  ...   \n",
       "7226             NM_000551.4(VHL):c.257C>T (p.Pro86Leu)               182977   \n",
       "7228             NM_000551.4(VHL):c.241C>T (p.Pro81Ser)                 2233   \n",
       "7231             NM_000551.4(VHL):c.154G>A (p.Glu52Lys)               161402   \n",
       "7418             NM_000551.4(VHL):c.263G>A (p.Trp88Ter)               182978   \n",
       "7862            NM_000546.6(TP53):c.105G>T (p.Leu35Phe)                12371   \n",
       "\n",
       "     Allele Registry Id                                   HGVS Expressions  \\\n",
       "82             CA000559  NM_000314.6:c.737C>T, NM_000314.5:c.737C>T, NM...   \n",
       "85             CA000358  NM_000314.6:c.235G>A, NM_000314.5:c.235G>A, NM...   \n",
       "94             CA000656  NM_000314.6:c.987_990delTAAA, NM_000314.5:c.98...   \n",
       "95             CA000219  NM_000314.6:c.892C>T, NC_000010.11:g.87960984C...   \n",
       "96             CA000235  NM_000314.6:c.964A>T, NC_000010.11:g.87961056A...   \n",
       "...                 ...                                                ...   \n",
       "7226           CA020186  NM_000551.4:c.257C>T, NC_000003.12:g.10142104C...   \n",
       "7228           CA020148  NM_000551.4:c.241C>T, NC_000003.12:g.10142088C...   \n",
       "7231           CA020056  NM_000551.4:c.154G>A, NC_000003.12:g.10142001G...   \n",
       "7418           CA020197  NM_000551.4:c.263G>A, NC_000003.12:g.10142110G...   \n",
       "7862           CA000025  NM_000546.6:c.105G>T, NC_000017.11:g.7676264C>...   \n",
       "\n",
       "     HGNC Gene Symbol                        Disease       Mondo Id  \\\n",
       "82               PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "85               PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "94               PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "95               PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "96               PTEN  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "...               ...                            ...            ...   \n",
       "7226              VHL      von Hippel-Lindau disease  MONDO:0008667   \n",
       "7228              VHL      von Hippel-Lindau disease  MONDO:0008667   \n",
       "7231              VHL      von Hippel-Lindau disease  MONDO:0008667   \n",
       "7418              VHL      von Hippel-Lindau disease  MONDO:0008667   \n",
       "7862             TP53           Li-Fraumeni syndrome  MONDO:0018875   \n",
       "\n",
       "                 Mode of Inheritance               Assertion  \\\n",
       "82    Autosomal dominant inheritance              Pathogenic   \n",
       "85    Autosomal dominant inheritance           Likely Benign   \n",
       "94    Autosomal dominant inheritance              Pathogenic   \n",
       "95    Autosomal dominant inheritance              Pathogenic   \n",
       "96    Autosomal dominant inheritance       Likely Pathogenic   \n",
       "...                              ...                     ...   \n",
       "7226  Autosomal dominant inheritance              Pathogenic   \n",
       "7228  Autosomal dominant inheritance                  Benign   \n",
       "7231  Autosomal dominant inheritance                  Benign   \n",
       "7418  Autosomal dominant inheritance              Pathogenic   \n",
       "7862  Autosomal dominant inheritance  Uncertain Significance   \n",
       "\n",
       "         Applied Evidence Codes (Met)  \\\n",
       "82     PM2, PS2_Very Strong, PS4, PP2   \n",
       "85           BS2_Supporting, BS1, PP2   \n",
       "94          PM2, PVS1, PS4_Supporting   \n",
       "95          PM2, PVS1, PS4_Supporting   \n",
       "96                          PVS1, PM2   \n",
       "...                               ...   \n",
       "7226  PM2_Supporting, PS4, PP1_Strong   \n",
       "7228                              BA1   \n",
       "7231                              BA1   \n",
       "7418        PVS1, PM2_Supporting, PS4   \n",
       "7862              PM2_Supporting, BP4   \n",
       "\n",
       "                       Applied Evidence Codes (Not Met)  \\\n",
       "82                                                  NaN   \n",
       "85                                                  NaN   \n",
       "94                                                  NaN   \n",
       "95                                                  NaN   \n",
       "96    PM6, BP2, BP4, BA1, PS4, PS2, PS1, PS3, BS4, B...   \n",
       "...                                                 ...   \n",
       "7226                                                NaN   \n",
       "7228                                                NaN   \n",
       "7231                                                PP3   \n",
       "7418                                                NaN   \n",
       "7862  PM5, PM1, PP3, PP4, PP1, BS2, BA1, PS1, PS3, P...   \n",
       "\n",
       "                              Summary of interpretation PubMed Articles  \\\n",
       "82    PTEN c.737C>T (p.P246L) meets criteria to be c...             NaN   \n",
       "85    PTEN c.235G>A (p.A79T) meets criteria to be cl...             NaN   \n",
       "94    PTEN c.987_990delTAAA (p.N329KfsX14) meets cri...             NaN   \n",
       "95    PTEN c.892C>T (p.Q298X) meets criteria to be c...             NaN   \n",
       "96    PTEN c.964A>T (p.Lys322Ter) meets criteria to ...             NaN   \n",
       "...                                                 ...             ...   \n",
       "7226  The variant NM_000551.4(VHL):c.257C>T (p.Pro86...             NaN   \n",
       "7228  The variant NM_000551.4(VHL):c.241C>T (p.Pro81...             NaN   \n",
       "7231  The variant NM_000551.3(VHL):c.154G>A (p.Glu52...             NaN   \n",
       "7418  The variant NM_000551.4(VHL):c.263G>A (p.Trp88...             NaN   \n",
       "7862  The NM_000546.6: c.105G>T variant in TP53 is a...             NaN   \n",
       "\n",
       "     Expert Panel Guideline Approval Date Published Date  Retracted  \\\n",
       "82      PTEN VCEP       NaN    2017-10-18     2018-12-10      False   \n",
       "85      PTEN VCEP       NaN    2018-04-06     2018-12-10      False   \n",
       "94      PTEN VCEP       NaN    2017-10-18     2018-12-10      False   \n",
       "95      PTEN VCEP       NaN    2017-10-18     2018-12-10      False   \n",
       "96      PTEN VCEP       NaN    2020-03-23     2020-03-26      False   \n",
       "...           ...       ...           ...            ...        ...   \n",
       "7226     VHL VCEP       NaN    2024-06-25     2024-06-25      False   \n",
       "7228     VHL VCEP       NaN    2024-06-25     2024-06-25      False   \n",
       "7231     VHL VCEP       NaN    2024-06-25     2024-06-25      False   \n",
       "7418     VHL VCEP       NaN    2024-06-25     2024-07-17      False   \n",
       "7862    TP53 VCEP       NaN    2024-09-06     2024-09-06      False   \n",
       "\n",
       "                                     Evidence Repo Link  \\\n",
       "82    https://erepo.genome.network/evrepo/ui/classif...   \n",
       "85    https://erepo.genome.network/evrepo/ui/classif...   \n",
       "94    https://erepo.genome.network/evrepo/ui/classif...   \n",
       "95    https://erepo.genome.network/evrepo/ui/classif...   \n",
       "96    https://erepo.genome.network/evrepo/ui/classif...   \n",
       "...                                                 ...   \n",
       "7226  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "7228  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "7231  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "7418  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "7862  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "\n",
       "                                      Uuid GeneSymbol  \\\n",
       "82    835a14e6-4a41-47df-a1ff-271c872adf39       PTEN   \n",
       "85    0340272b-ab4c-4672-94b8-40e0525d8eda       PTEN   \n",
       "94    e41ad782-4c9c-4ae6-ba02-ec2e11c967fc       PTEN   \n",
       "95    f890d81d-90ed-4411-a0fd-a89669dd616a       PTEN   \n",
       "96    bfdfbf81-6f26-43f8-aa4b-bbb17ebaf54a       PTEN   \n",
       "...                                    ...        ...   \n",
       "7226  318690ae-2743-4195-8d9b-e2e84ff94494        VHL   \n",
       "7228  416afbe6-3dd6-40d9-9e61-d3c54b017b5e        VHL   \n",
       "7231  9f120c06-bb61-4115-9c9c-187ef481f769        VHL   \n",
       "7418  eb3883a6-80f5-45f8-9c0f-a2928a4e24ca        VHL   \n",
       "7862  b73a437c-26a5-4491-8b92-edbe32570d9e       TP53   \n",
       "\n",
       "                            metEvidence  \\\n",
       "82     [PM2, PS2_Very Strong, PS4, PP2]   \n",
       "85           [BS2_Supporting, BS1, PP2]   \n",
       "94          [PM2, PVS1, PS4_Supporting]   \n",
       "95          [PM2, PVS1, PS4_Supporting]   \n",
       "96                          [PVS1, PM2]   \n",
       "...                                 ...   \n",
       "7226  [PM2_Supporting, PS4, PP1_Strong]   \n",
       "7228                              [BA1]   \n",
       "7231                              [BA1]   \n",
       "7418        [PVS1, PM2_Supporting, PS4]   \n",
       "7862              [PM2_Supporting, BP4]   \n",
       "\n",
       "                                            Expressions  \\\n",
       "82    (NM_000314.6:c.737C>T, NM_000314.5:c.737C>T, N...   \n",
       "85    (NM_000314.6:c.235G>A, NM_000314.5:c.235G>A, N...   \n",
       "94    (NM_000314.6:c.987_990delTAAA, NM_000314.5:c.9...   \n",
       "95    (NM_000314.6:c.892C>T, NC_000010.11:g.87960984...   \n",
       "96    (NM_000314.6:c.964A>T, NC_000010.11:g.87961056...   \n",
       "...                                                 ...   \n",
       "7226  (NM_000551.4:c.257C>T, NC_000003.12:g.10142104...   \n",
       "7228  (NM_000551.4:c.241C>T, NC_000003.12:g.10142088...   \n",
       "7231  (NM_000551.4:c.154G>A, NC_000003.12:g.10142001...   \n",
       "7418  (NM_000551.4:c.263G>A, NC_000003.12:g.10142110...   \n",
       "7862  (NM_000546.6:c.105G>T, NC_000017.11:g.7676264C...   \n",
       "\n",
       "                                          hgvs_nuc_full  \\\n",
       "82    (NM_001304717.2:c.1256C>T, NM_000314.5:c.737C>...   \n",
       "85    (NM_001304717.5:c.754G>A, NM_000314.7:c.235G>A...   \n",
       "94                                                   ()   \n",
       "95    (NM_001304718.2:c.301C>T, NM_001304718.1:c.301...   \n",
       "96    (NM_001304717.5:c.1483A>T, NM_001304717.2:c.14...   \n",
       "...                                                 ...   \n",
       "7226  (NM_198156.3:c.257C>T, NM_198156.2:c.257C>T, N...   \n",
       "7228  (NM_000551.3:c.241C>T, NM_001354723.2:c.241C>T...   \n",
       "7231  (NM_198156.3:c.154G>A, NM_000551.4:c.154G>A, N...   \n",
       "7418  (NM_198156.2:c.263G>A, NM_001354723.1:c.263G>A...   \n",
       "7862  (NM_000546.6:c.105G>T, NM_001126114.2:c.105G>T...   \n",
       "\n",
       "                                               hgvs_nuc     hgvs_pro  \n",
       "82    (c.692C>T, c.641C>T, c.1256C>T, c.737C>T, c.14...  p.Pro246Leu  \n",
       "85    (c.235G>A, c.133G>A, c.190G>A, c.754G>A, c.139...   p.Ala79Thr  \n",
       "94                                                   ()  p.Asn329Lys  \n",
       "95    (c.796C>T, c.1411C>T, c.892C>T, c.301C>T, c.84...  p.Gln298Ter  \n",
       "96    (c.919A>T, c.964A>T, c.868A>T, c.1483A>T, c.37...  p.Lys322Ter  \n",
       "...                                                 ...          ...  \n",
       "7226                                        (c.257C>T,)   p.Pro86Leu  \n",
       "7228                                        (c.241C>T,)   p.Pro81Ser  \n",
       "7231                                        (c.154G>A,)   p.Glu52Lys  \n",
       "7418                                        (c.263G>A,)   p.Trp88Ter  \n",
       "7862                                        (c.105G>T,)   p.Leu35Phe  \n",
       "\n",
       "[66 rows x 26 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "evidenceDF[(evidenceDF.loc[:,'#Variation'].isin(set(df.loc[:,'#Variation'].values))) & \\\n",
    "           np.logical_not(evidenceDF.loc[:,'Applied Evidence Codes (Met)'].str.contains('BS3').values) & \\\n",
    "            np.logical_not(evidenceDF.loc[:,'Applied Evidence Codes (Met)'].str.contains('PS3').values)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "assign_assay_evidence_strength(-2.2, thresholdsP['Kato_TP53'], thresholdsB['Kato_TP53'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "def assign_assay_evidence_strength(score, pathogenic_score_thresholds, benign_score_thresholds):\n",
    "    is_inverted = (pathogenic_score_thresholds[0] > benign_score_thresholds[0]) or (pathogenic_score_thresholds[0] < pathogenic_score_thresholds[1])\n",
    "    if np.isnan(score):\n",
    "        return 0\n",
    "    for threshold,points in list(zip(pathogenic_score_thresholds,[1,2,3,4,5,6,7,8]))[::-1]:\n",
    "        if np.isnan(threshold):\n",
    "            continue\n",
    "        if is_inverted and score >= threshold:\n",
    "            return points\n",
    "        if (not is_inverted) and score <= threshold:\n",
    "            return points\n",
    "    for threshold,points in list(zip(benign_score_thresholds,[-1,-2,-3,-4,-5,-6,-7,-8]))[::-1]:\n",
    "        if np.isnan(threshold):\n",
    "            continue\n",
    "        if is_inverted and score <= threshold:\n",
    "            return points\n",
    "        if (not is_inverted) and score >= threshold:\n",
    "            return points\n",
    "    return 0\n",
    "\n",
    "thresholdsP = {}\n",
    "thresholdsB = {}\n",
    "for dsname in df.scoreset_source.unique():\n",
    "    pipeline = datasets[dsname]['pipeline']\n",
    "    tauP = np.load(f\"/data/dzeiberg/mave_calibration/figs_10_09_24/{dsname}_pipeline_{pipeline}/Tau_p.npy\")\n",
    "    tauB = np.load(f\"/data/dzeiberg/mave_calibration/figs_10_09_24/{dsname}_pipeline_{pipeline}/Tau_b.npy\")\n",
    "    thresholdsP[dsname] = tauP\n",
    "    thresholdsB[dsname] = tauB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "strengths = []\n",
    "for idx,r in df.iterrows():\n",
    "    strength = assign_assay_evidence_strength(r.avg_score, thresholdsP[r.scoreset_source], thresholdsB[r.scoreset_source])\n",
    "    strengths.append(strength)\n",
    "df = df.assign(strength=strengths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.assign(strengthAbs=df.strength.abs()).sort_values(by='strengthAbs',ascending=False)\n",
    "dfMax = df.drop_duplicates(subset=['hgvs_pro','GeneSymbol'],keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(251, 29)"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneSymbol\n",
       "PTEN     114\n",
       "TP53     109\n",
       "VHL       18\n",
       "BRCA2      5\n",
       "BRCA1      5\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax.GeneSymbol.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "scoreset_source\n",
       "Kato_TP53                95\n",
       "Mighell_PTEN             75\n",
       "Matreyek_PTEN_VampSeq    39\n",
       "Buckley_VHL_SGE          18\n",
       "Giacomelli_3              9\n",
       "Hu_BRCA2_HDR              5\n",
       "Findlay_BRCA1_SGE         4\n",
       "Giacomelli_1              4\n",
       "Giacomelli_2              1\n",
       "Starita_BRCA1_HDR         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax.scoreset_source.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>strength</th>\n",
       "      <th>-6</th>\n",
       "      <th>-5</th>\n",
       "      <th>-4</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assertion</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Benign</th>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Likely Benign</th>\n",
       "      <td>44</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Likely Pathogenic</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uncertain Significance</th>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "strength                -6  -5  -4   0   1   2   5   6   7   8\n",
       "Assertion                                                     \n",
       "Benign                  15   0   0   3   0   0   0   0   0   0\n",
       "Likely Benign           44   3   0   5   0   0   0   1   0   0\n",
       "Likely Pathogenic        0   0   0  22   0   1   1   5  12   2\n",
       "Pathogenic               1   1   0  33   1   0   1  21  11  13\n",
       "Uncertain Significance  12   2   1  32   0   0   0   6   2   0"
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax.groupby(\"Assertion\").strength.value_counts().unstack().fillna(0).astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "strength\n",
       "2     1\n",
       "5     1\n",
       "6    25\n",
       "7    16\n",
       "8    10\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 251,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax[(dfMax.strength >= 1) & (dfMax.loc[:,'Applied Evidence Codes (Met)'].str.contains(\"PS3\"))].strength.value_counts().sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(91, 29)"
      ]
     },
     "execution_count": 242,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax[(dfMax.loc[:,'Applied Evidence Codes (Met)'].str.contains(\"PS3\"))].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>avg_score</th>\n",
       "      <th>#Variation</th>\n",
       "      <th>ClinVar Variation Id</th>\n",
       "      <th>Allele Registry Id</th>\n",
       "      <th>HGVS Expressions</th>\n",
       "      <th>HGNC Gene Symbol</th>\n",
       "      <th>Disease</th>\n",
       "      <th>Mondo Id</th>\n",
       "      <th>Mode of Inheritance</th>\n",
       "      <th>Assertion</th>\n",
       "      <th>Applied Evidence Codes (Met)</th>\n",
       "      <th>Applied Evidence Codes (Not Met)</th>\n",
       "      <th>Summary of interpretation</th>\n",
       "      <th>PubMed Articles</th>\n",
       "      <th>Expert Panel</th>\n",
       "      <th>Guideline</th>\n",
       "      <th>Approval Date</th>\n",
       "      <th>Published Date</th>\n",
       "      <th>Retracted</th>\n",
       "      <th>Evidence Repo Link</th>\n",
       "      <th>Uuid</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>metEvidence</th>\n",
       "      <th>Expressions</th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>scoreset_source</th>\n",
       "      <th>strength</th>\n",
       "      <th>strengthAbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>p.Tyr27Ser</td>\n",
       "      <td>0.508829</td>\n",
       "      <td>NM_000314.8(PTEN):c.80A&gt;C (p.Tyr27Ser)</td>\n",
       "      <td>280724</td>\n",
       "      <td>CA10603162</td>\n",
       "      <td>NM_000314.8:c.80A&gt;C, NC_000010.11:g.87894025A&gt;...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PS2, PS3_Supporting, PP2, PP3, PS4_Supporting,...</td>\n",
       "      <td>BS3</td>\n",
       "      <td>NM_000314.8(PTEN):c.80A&gt;C (p.Tyr27Ser) meets c...</td>\n",
       "      <td>10866302</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2023-08-04</td>\n",
       "      <td>2023-10-19</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>1d6c53d1-6cb2-4e76-a5e8-dcc135e646ff</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PS2, PS3_Supporting, PP2, PP3, PS4_Supporting...</td>\n",
       "      <td>(NM_000314.8:c.80A&gt;C, NC_000010.11:g.87894025A...</td>\n",
       "      <td>p.Tyr27Ser</td>\n",
       "      <td>Matreyek_PTEN_VampSeq</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>p.Leu188Val</td>\n",
       "      <td>-0.391271</td>\n",
       "      <td>NM_000551.4(VHL):c.562C&gt;G (p.Leu188Val)</td>\n",
       "      <td>2225</td>\n",
       "      <td>CA020488</td>\n",
       "      <td>NM_000551.4:c.562C&gt;G, NC_000003.12:g.10149885C...</td>\n",
       "      <td>VHL</td>\n",
       "      <td>von Hippel-Lindau disease</td>\n",
       "      <td>MONDO:0008667</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>PS3_Supporting, PP3, BS2, PM1, BS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The variant NM_000551.4(VHL):c.562C&gt;G (p.Leu18...</td>\n",
       "      <td>15574766, 17700531, 16452184, 12097293, 190302...</td>\n",
       "      <td>VHL VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>03093ff9-64cb-4fd3-a97b-d60308334b8c</td>\n",
       "      <td>VHL</td>\n",
       "      <td>[PS3_Supporting, PP3, BS2, PM1, BS1]</td>\n",
       "      <td>(NM_000551.4:c.562C&gt;G, NC_000003.12:g.10149885...</td>\n",
       "      <td>p.Leu188Val</td>\n",
       "      <td>Buckley_VHL_SGE</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>p.Phe91Leu</td>\n",
       "      <td>-0.137526</td>\n",
       "      <td>NM_000551.4(VHL):c.273C&gt;A (p.Phe91Leu)</td>\n",
       "      <td>411978</td>\n",
       "      <td>CA16611270</td>\n",
       "      <td>NM_000551.4:c.273C&gt;A, NC_000003.12:g.10142120C...</td>\n",
       "      <td>VHL</td>\n",
       "      <td>von Hippel-Lindau disease</td>\n",
       "      <td>MONDO:0008667</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>PS3_Supporting, PP3, PM2_Supporting, PM1</td>\n",
       "      <td>PS4</td>\n",
       "      <td>The NM_000551.3(VHL):c.273C&gt;A (p.Phe91Leu) var...</td>\n",
       "      <td>31337753</td>\n",
       "      <td>VHL VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>d6ad76b6-e4e0-490b-8d57-e1a11cc12139</td>\n",
       "      <td>VHL</td>\n",
       "      <td>[PS3_Supporting, PP3, PM2_Supporting, PM1]</td>\n",
       "      <td>(NM_000551.4:c.273C&gt;A, NC_000003.12:g.10142120...</td>\n",
       "      <td>p.Phe91Leu</td>\n",
       "      <td>Buckley_VHL_SGE</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>p.Met35Val</td>\n",
       "      <td>1.061209</td>\n",
       "      <td>NM_000314.6(PTEN):c.103A&gt;G (p.Met35Val)</td>\n",
       "      <td>231916</td>\n",
       "      <td>CA10578906</td>\n",
       "      <td>NM_000314.6:c.103A&gt;G, NC_000010.11:g.87894048A...</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>PTEN hamartoma tumor syndrome</td>\n",
       "      <td>MONDO:0017623</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PS2_Very Strong, PS3_Supporting, PP3, PP2, PM2...</td>\n",
       "      <td>PM6, PVS1, BA1, BP4, BP1, BP2, BP3, PS1, BP5, ...</td>\n",
       "      <td>NM_000314.8(PTEN):c.103A&gt;G (p.Met35Val) meets ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-09</td>\n",
       "      <td>2024-03-04</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>4f6345e3-25b1-48cd-9697-1a5624715071</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>[PS2_Very Strong, PS3_Supporting, PP3, PP2, PM...</td>\n",
       "      <td>(NM_000314.6:c.103A&gt;G, NC_000010.11:g.87894048...</td>\n",
       "      <td>p.Met35Val</td>\n",
       "      <td>Matreyek_PTEN_VampSeq</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>p.Arg64Pro</td>\n",
       "      <td>-0.477015</td>\n",
       "      <td>NM_000551.4(VHL):c.191G&gt;C (p.Arg64Pro)</td>\n",
       "      <td>2226</td>\n",
       "      <td>CA020089</td>\n",
       "      <td>NM_000551.4:c.191G&gt;C, NC_000003.12:g.10142038G...</td>\n",
       "      <td>VHL</td>\n",
       "      <td>von Hippel-Lindau disease</td>\n",
       "      <td>MONDO:0008667</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PS3_Supporting, PP3, PM2_Supporting, PS4, PM6_...</td>\n",
       "      <td>PP1</td>\n",
       "      <td>The NM_000551.3(VHL):c.191G&gt;C (p.Arg64Pro) var...</td>\n",
       "      <td>16452184, 11331612, 15611064</td>\n",
       "      <td>VHL VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-25</td>\n",
       "      <td>2024-07-17</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classif...</td>\n",
       "      <td>7122665e-2b75-4e58-a96a-965f11cf866f</td>\n",
       "      <td>VHL</td>\n",
       "      <td>[PS3_Supporting, PP3, PM2_Supporting, PS4, PM6...</td>\n",
       "      <td>(NM_000551.4:c.191G&gt;C, NC_000003.12:g.10142038...</td>\n",
       "      <td>p.Arg64Pro</td>\n",
       "      <td>Buckley_VHL_SGE</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hgvs_pro  avg_score                               #Variation  \\\n",
       "51   p.Tyr27Ser   0.508829   NM_000314.8(PTEN):c.80A>C (p.Tyr27Ser)   \n",
       "17  p.Leu188Val  -0.391271  NM_000551.4(VHL):c.562C>G (p.Leu188Val)   \n",
       "12   p.Phe91Leu  -0.137526   NM_000551.4(VHL):c.273C>A (p.Phe91Leu)   \n",
       "62   p.Met35Val   1.061209  NM_000314.6(PTEN):c.103A>G (p.Met35Val)   \n",
       "1    p.Arg64Pro  -0.477015   NM_000551.4(VHL):c.191G>C (p.Arg64Pro)   \n",
       "\n",
       "   ClinVar Variation Id Allele Registry Id  \\\n",
       "51               280724         CA10603162   \n",
       "17                 2225           CA020488   \n",
       "12               411978         CA16611270   \n",
       "62               231916         CA10578906   \n",
       "1                  2226           CA020089   \n",
       "\n",
       "                                     HGVS Expressions HGNC Gene Symbol  \\\n",
       "51  NM_000314.8:c.80A>C, NC_000010.11:g.87894025A>...             PTEN   \n",
       "17  NM_000551.4:c.562C>G, NC_000003.12:g.10149885C...              VHL   \n",
       "12  NM_000551.4:c.273C>A, NC_000003.12:g.10142120C...              VHL   \n",
       "62  NM_000314.6:c.103A>G, NC_000010.11:g.87894048A...             PTEN   \n",
       "1   NM_000551.4:c.191G>C, NC_000003.12:g.10142038G...              VHL   \n",
       "\n",
       "                          Disease       Mondo Id  \\\n",
       "51  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "17      von Hippel-Lindau disease  MONDO:0008667   \n",
       "12      von Hippel-Lindau disease  MONDO:0008667   \n",
       "62  PTEN hamartoma tumor syndrome  MONDO:0017623   \n",
       "1       von Hippel-Lindau disease  MONDO:0008667   \n",
       "\n",
       "               Mode of Inheritance               Assertion  \\\n",
       "51  Autosomal dominant inheritance       Likely Pathogenic   \n",
       "17  Autosomal dominant inheritance  Uncertain Significance   \n",
       "12  Autosomal dominant inheritance  Uncertain Significance   \n",
       "62  Autosomal dominant inheritance              Pathogenic   \n",
       "1   Autosomal dominant inheritance              Pathogenic   \n",
       "\n",
       "                         Applied Evidence Codes (Met)  \\\n",
       "51  PS2, PS3_Supporting, PP2, PP3, PS4_Supporting,...   \n",
       "17                 PS3_Supporting, PP3, BS2, PM1, BS1   \n",
       "12           PS3_Supporting, PP3, PM2_Supporting, PM1   \n",
       "62  PS2_Very Strong, PS3_Supporting, PP3, PP2, PM2...   \n",
       "1   PS3_Supporting, PP3, PM2_Supporting, PS4, PM6_...   \n",
       "\n",
       "                     Applied Evidence Codes (Not Met)  \\\n",
       "51                                                BS3   \n",
       "17                                                NaN   \n",
       "12                                                PS4   \n",
       "62  PM6, PVS1, BA1, BP4, BP1, BP2, BP3, PS1, BP5, ...   \n",
       "1                                                 PP1   \n",
       "\n",
       "                            Summary of interpretation  \\\n",
       "51  NM_000314.8(PTEN):c.80A>C (p.Tyr27Ser) meets c...   \n",
       "17  The variant NM_000551.4(VHL):c.562C>G (p.Leu18...   \n",
       "12  The NM_000551.3(VHL):c.273C>A (p.Phe91Leu) var...   \n",
       "62  NM_000314.8(PTEN):c.103A>G (p.Met35Val) meets ...   \n",
       "1   The NM_000551.3(VHL):c.191G>C (p.Arg64Pro) var...   \n",
       "\n",
       "                                      PubMed Articles Expert Panel Guideline  \\\n",
       "51                                           10866302    PTEN VCEP       NaN   \n",
       "17  15574766, 17700531, 16452184, 12097293, 190302...     VHL VCEP       NaN   \n",
       "12                                           31337753     VHL VCEP       NaN   \n",
       "62                                                NaN    PTEN VCEP       NaN   \n",
       "1                        16452184, 11331612, 15611064     VHL VCEP       NaN   \n",
       "\n",
       "   Approval Date Published Date  Retracted  \\\n",
       "51    2023-08-04     2023-10-19      False   \n",
       "17    2024-06-25     2024-06-25      False   \n",
       "12    2024-06-25     2024-06-25      False   \n",
       "62    2024-02-09     2024-03-04      False   \n",
       "1     2024-06-25     2024-07-17      False   \n",
       "\n",
       "                                   Evidence Repo Link  \\\n",
       "51  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "17  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "12  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "62  https://erepo.genome.network/evrepo/ui/classif...   \n",
       "1   https://erepo.genome.network/evrepo/ui/classif...   \n",
       "\n",
       "                                    Uuid GeneSymbol  \\\n",
       "51  1d6c53d1-6cb2-4e76-a5e8-dcc135e646ff       PTEN   \n",
       "17  03093ff9-64cb-4fd3-a97b-d60308334b8c        VHL   \n",
       "12  d6ad76b6-e4e0-490b-8d57-e1a11cc12139        VHL   \n",
       "62  4f6345e3-25b1-48cd-9697-1a5624715071       PTEN   \n",
       "1   7122665e-2b75-4e58-a96a-965f11cf866f        VHL   \n",
       "\n",
       "                                          metEvidence  \\\n",
       "51  [PS2, PS3_Supporting, PP2, PP3, PS4_Supporting...   \n",
       "17               [PS3_Supporting, PP3, BS2, PM1, BS1]   \n",
       "12         [PS3_Supporting, PP3, PM2_Supporting, PM1]   \n",
       "62  [PS2_Very Strong, PS3_Supporting, PP3, PP2, PM...   \n",
       "1   [PS3_Supporting, PP3, PM2_Supporting, PS4, PM6...   \n",
       "\n",
       "                                          Expressions     hgvs_pro  \\\n",
       "51  (NM_000314.8:c.80A>C, NC_000010.11:g.87894025A...   p.Tyr27Ser   \n",
       "17  (NM_000551.4:c.562C>G, NC_000003.12:g.10149885...  p.Leu188Val   \n",
       "12  (NM_000551.4:c.273C>A, NC_000003.12:g.10142120...   p.Phe91Leu   \n",
       "62  (NM_000314.6:c.103A>G, NC_000010.11:g.87894048...   p.Met35Val   \n",
       "1   (NM_000551.4:c.191G>C, NC_000003.12:g.10142038...   p.Arg64Pro   \n",
       "\n",
       "          scoreset_source  strength  strengthAbs  \n",
       "51  Matreyek_PTEN_VampSeq         0            0  \n",
       "17        Buckley_VHL_SGE         0            0  \n",
       "12        Buckley_VHL_SGE         0            0  \n",
       "62  Matreyek_PTEN_VampSeq         0            0  \n",
       "1         Buckley_VHL_SGE         0            0  "
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax[(dfMax.strength < 1) & (dfMax.loc[:,'Applied Evidence Codes (Met)'].str.contains(\"PS3\")) & \\\n",
    "    (dfMax.scoreset_source != \"Mighell_PTEN\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Summary of interpretation</th>\n",
       "      <th>Applied Evidence Codes (Met)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>NM_000314.8(PTEN):c.80A&gt;C (p.Tyr27Ser) meets c...</td>\n",
       "      <td>PS2, PS3_Supporting, PP2, PP3, PS4_Supporting,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PTEN c.16A&gt;G (p.Lys6Glu) meets criteria to be ...</td>\n",
       "      <td>PS2_Very Strong, PM2, PS4_Moderate, PS3_Suppor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NM_000314.8(PTEN):c.40A&gt;G (p.Arg14Gly) is curr...</td>\n",
       "      <td>PS3_Supporting, PP2, PP3, PM2_Supporting, PS4_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NM_000314.8(PTEN):c.44G&gt;C (p.Arg15Thr) meets c...</td>\n",
       "      <td>PS2, PP2, PP3, PS3_Moderate, PM2_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NM_000314.8(PTEN):c.44G&gt;A (p.Arg15Lys) meets c...</td>\n",
       "      <td>PP2, PS3_Moderate, PM6_Strong, PM2_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NM_000314.8(PTEN):c.49C&gt;G (p.Gln17Glu) is curr...</td>\n",
       "      <td>PM6, BP4, PS3_Supporting, PP2, PM2_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NM_000314.8(PTEN):c.106G&gt;A (p.Gly36Arg) meets ...</td>\n",
       "      <td>PS4_Moderate, PP1, PP2, PP3, PM2_Supporting, P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NM_000314.8(PTEN):c.107G&gt;A (p.Gly36Glu) meets ...</td>\n",
       "      <td>PS2, PP2, PP3, PM2_Supporting, PS3_Moderate, P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NM_000314.8(PTEN):c.113C&gt;T (p.Pro38Leu) meets ...</td>\n",
       "      <td>PS2_Very Strong, PP2, PP3, PS3_Moderate, PS4_S...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NM_000314.8(PTEN):c.112C&gt;T (p.Pro38Ser) is cur...</td>\n",
       "      <td>PP2, PP3, PM2_Supporting, PS3_Moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NM_000314.8(PTEN):c.140G&gt;A (p.Arg47Lys) meets ...</td>\n",
       "      <td>PP2, PP3, PM6_Strong, PS3_Moderate, PM2_Suppor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NM_000314.8(PTEN):c.149T&gt;C (p.Ile50Thr) meets ...</td>\n",
       "      <td>PP2, PP3, PM2_Supporting, PS3_Moderate, PM6_St...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>PTEN c.209T&gt;C (p.Leu70Pro) meets criteria to b...</td>\n",
       "      <td>PM2, PS3, PP2, PS4_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NM_000314.8(PTEN):c.212G&gt;A (p.Cys71Tyr) meets ...</td>\n",
       "      <td>PS2_Very Strong, PP2, PP3, PM2_Supporting, PS3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NM_000314.8(PTEN):c.278A&gt;C (p.His93Pro) varian...</td>\n",
       "      <td>PM5, PM1, PP2, PP3, PM2_Supporting, PS3_Moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>PTEN c.278A&gt;G (p.His93Arg) meets criteria to b...</td>\n",
       "      <td>PM2, PS2, PM1, PS3_Supporting, PP2, PS4_Suppor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>PTEN c.284C&gt;T (p.Pro95Leu) meets criteria to b...</td>\n",
       "      <td>PM6, PM2, PS3, PP2, PS4_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>PTEN c.367C&gt;G (p.His123Asp) meets criteria to ...</td>\n",
       "      <td>PM2, PS3, PM1, PP2, PS4_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>PTEN c.395G&gt;A (p.Gly132Asp) meets criteria to ...</td>\n",
       "      <td>PM2, PM6, PS4, PS3, PP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>PTEN c.389G&gt;A (p.R130Q) meets criteria to be c...</td>\n",
       "      <td>PM2, PS4, PS3, PM1, PP2, PM6_Strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>PTEN c.389G&gt;C (p.Arg130Pro) meets criteria to ...</td>\n",
       "      <td>PM2, PS3, PM1, PP2, PS4_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>PTEN c.388C&gt;G (p.Arg130Gly) meets criteria to ...</td>\n",
       "      <td>PM2, PS3, PM5, PM1, PP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>PTEN c.517C&gt;T (p.R173C) meets criteria to be c...</td>\n",
       "      <td>PM2, PS2_Very Strong, PS4_Moderate, PS3, PP2, PP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>PTEN c.520T&gt;A (p.Tyr174Asn) meets criteria to ...</td>\n",
       "      <td>PM2, PS3, PP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>PTEN c.407G&gt;A (p.C136Y) meets criteria to be c...</td>\n",
       "      <td>PM2, PS3, PS2, PP2, PS4_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>PTEN c.392C&gt;T (p.Thr131Ile) meets criteria to ...</td>\n",
       "      <td>PM2, PS3, PS4_Moderate, PP2, PM6_Strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>PTEN c.464A&gt;G (p.Tyr155Cys) meets criteria to ...</td>\n",
       "      <td>PM6, PM2, PS4, PS3, PP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>PTEN c.477G&gt;T (p.Arg159Ser) meets criteria to ...</td>\n",
       "      <td>PM2, PS3, PP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>PTEN c.493G&gt;A (p.Gly165Arg) meets criteria to ...</td>\n",
       "      <td>PM2, PS2, PS3, PP2, PS4_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>PTEN c.740T&gt;C (p.Leu247Ser) meets criteria to ...</td>\n",
       "      <td>PP2, PP3, PS3_Moderate, PM6_Strong, PM2_Suppor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NM_000314.8(PTEN):c.830C&gt;T (p.Thr277Ile) varia...</td>\n",
       "      <td>PP2, PP3, PM2_Supporting, PS3_Moderate, PM6_St...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>PTEN c.542T&gt;C (p.Leu181Pro) meets criteria to ...</td>\n",
       "      <td>PM2, PS3, PS2, PP2, PS4_Supporting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>PTEN c.610C&gt;G (p.Pro204Ala) meets criteria to ...</td>\n",
       "      <td>PP2, PP3, PM6_Strong, PS3_Moderate, PM2_Suppor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>NM_000314.8(PTEN):c.947T&gt;C (p.Leu316Pro) meets...</td>\n",
       "      <td>PM6, PP2, PP3, PS4_Supporting, PM2_Supporting,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>NM_000314.8(PTEN):c.103A&gt;G (p.Met35Val) meets ...</td>\n",
       "      <td>PS2_Very Strong, PS3_Supporting, PP3, PP2, PM2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Summary of interpretation  \\\n",
       "51   NM_000314.8(PTEN):c.80A>C (p.Tyr27Ser) meets c...   \n",
       "2    PTEN c.16A>G (p.Lys6Glu) meets criteria to be ...   \n",
       "4    NM_000314.8(PTEN):c.40A>G (p.Arg14Gly) is curr...   \n",
       "6    NM_000314.8(PTEN):c.44G>C (p.Arg15Thr) meets c...   \n",
       "7    NM_000314.8(PTEN):c.44G>A (p.Arg15Lys) meets c...   \n",
       "10   NM_000314.8(PTEN):c.49C>G (p.Gln17Glu) is curr...   \n",
       "19   NM_000314.8(PTEN):c.106G>A (p.Gly36Arg) meets ...   \n",
       "21   NM_000314.8(PTEN):c.107G>A (p.Gly36Glu) meets ...   \n",
       "22   NM_000314.8(PTEN):c.113C>T (p.Pro38Leu) meets ...   \n",
       "23   NM_000314.8(PTEN):c.112C>T (p.Pro38Ser) is cur...   \n",
       "24   NM_000314.8(PTEN):c.140G>A (p.Arg47Lys) meets ...   \n",
       "25   NM_000314.8(PTEN):c.149T>C (p.Ile50Thr) meets ...   \n",
       "31   PTEN c.209T>C (p.Leu70Pro) meets criteria to b...   \n",
       "32   NM_000314.8(PTEN):c.212G>A (p.Cys71Tyr) meets ...   \n",
       "35   NM_000314.8(PTEN):c.278A>C (p.His93Pro) varian...   \n",
       "36   PTEN c.278A>G (p.His93Arg) meets criteria to b...   \n",
       "37   PTEN c.284C>T (p.Pro95Leu) meets criteria to b...   \n",
       "51   PTEN c.367C>G (p.His123Asp) meets criteria to ...   \n",
       "57   PTEN c.395G>A (p.Gly132Asp) meets criteria to ...   \n",
       "53   PTEN c.389G>A (p.R130Q) meets criteria to be c...   \n",
       "55   PTEN c.389G>C (p.Arg130Pro) meets criteria to ...   \n",
       "54   PTEN c.388C>G (p.Arg130Gly) meets criteria to ...   \n",
       "71   PTEN c.517C>T (p.R173C) meets criteria to be c...   \n",
       "72   PTEN c.520T>A (p.Tyr174Asn) meets criteria to ...   \n",
       "59   PTEN c.407G>A (p.C136Y) meets criteria to be c...   \n",
       "56   PTEN c.392C>T (p.Thr131Ile) meets criteria to ...   \n",
       "62   PTEN c.464A>G (p.Tyr155Cys) meets criteria to ...   \n",
       "64   PTEN c.477G>T (p.Arg159Ser) meets criteria to ...   \n",
       "65   PTEN c.493G>A (p.Gly165Arg) meets criteria to ...   \n",
       "90   PTEN c.740T>C (p.Leu247Ser) meets criteria to ...   \n",
       "95   NM_000314.8(PTEN):c.830C>T (p.Thr277Ile) varia...   \n",
       "75   PTEN c.542T>C (p.Leu181Pro) meets criteria to ...   \n",
       "80   PTEN c.610C>G (p.Pro204Ala) meets criteria to ...   \n",
       "104  NM_000314.8(PTEN):c.947T>C (p.Leu316Pro) meets...   \n",
       "62   NM_000314.8(PTEN):c.103A>G (p.Met35Val) meets ...   \n",
       "\n",
       "                          Applied Evidence Codes (Met)  \n",
       "51   PS2, PS3_Supporting, PP2, PP3, PS4_Supporting,...  \n",
       "2    PS2_Very Strong, PM2, PS4_Moderate, PS3_Suppor...  \n",
       "4    PS3_Supporting, PP2, PP3, PM2_Supporting, PS4_...  \n",
       "6          PS2, PP2, PP3, PS3_Moderate, PM2_Supporting  \n",
       "7        PP2, PS3_Moderate, PM6_Strong, PM2_Supporting  \n",
       "10       PM6, BP4, PS3_Supporting, PP2, PM2_Supporting  \n",
       "19   PS4_Moderate, PP1, PP2, PP3, PM2_Supporting, P...  \n",
       "21   PS2, PP2, PP3, PM2_Supporting, PS3_Moderate, P...  \n",
       "22   PS2_Very Strong, PP2, PP3, PS3_Moderate, PS4_S...  \n",
       "23              PP2, PP3, PM2_Supporting, PS3_Moderate  \n",
       "24   PP2, PP3, PM6_Strong, PS3_Moderate, PM2_Suppor...  \n",
       "25   PP2, PP3, PM2_Supporting, PS3_Moderate, PM6_St...  \n",
       "31                       PM2, PS3, PP2, PS4_Supporting  \n",
       "32   PS2_Very Strong, PP2, PP3, PM2_Supporting, PS3...  \n",
       "35    PM5, PM1, PP2, PP3, PM2_Supporting, PS3_Moderate  \n",
       "36   PM2, PS2, PM1, PS3_Supporting, PP2, PS4_Suppor...  \n",
       "37                  PM6, PM2, PS3, PP2, PS4_Supporting  \n",
       "51                  PM2, PS3, PM1, PP2, PS4_Supporting  \n",
       "57                             PM2, PM6, PS4, PS3, PP2  \n",
       "53                 PM2, PS4, PS3, PM1, PP2, PM6_Strong  \n",
       "55                  PM2, PS3, PM1, PP2, PS4_Supporting  \n",
       "54                             PM2, PS3, PM5, PM1, PP2  \n",
       "71   PM2, PS2_Very Strong, PS4_Moderate, PS3, PP2, PP1  \n",
       "72                                       PM2, PS3, PP2  \n",
       "59                  PM2, PS3, PS2, PP2, PS4_Supporting  \n",
       "56             PM2, PS3, PS4_Moderate, PP2, PM6_Strong  \n",
       "62                             PM6, PM2, PS4, PS3, PP2  \n",
       "64                                       PM2, PS3, PP2  \n",
       "65                  PM2, PS2, PS3, PP2, PS4_Supporting  \n",
       "90   PP2, PP3, PS3_Moderate, PM6_Strong, PM2_Suppor...  \n",
       "95   PP2, PP3, PM2_Supporting, PS3_Moderate, PM6_St...  \n",
       "75                  PM2, PS3, PS2, PP2, PS4_Supporting  \n",
       "80   PP2, PP3, PM6_Strong, PS3_Moderate, PM2_Suppor...  \n",
       "104  PM6, PP2, PP3, PS4_Supporting, PM2_Supporting,...  \n",
       "62   PS2_Very Strong, PS3_Supporting, PP3, PP2, PM2...  "
      ]
     },
     "execution_count": 266,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax[(dfMax.strength < 1) & \\\n",
    "    (dfMax.loc[:,'Applied Evidence Codes (Met)'].str.contains(\"PS3\")) & \\\n",
    "        (dfMax.GeneSymbol == \"PTEN\")].loc[:,[\"Summary of interpretation\",'Applied Evidence Codes (Met)']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [],
   "source": [
    "import textwrap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Summary of interpretation</th>\n",
       "      <th>scoreset_source</th>\n",
       "      <th>strength</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>The c.5509T&gt;G variant in BRCA1 is a missense v...</td>\n",
       "      <td>Findlay_BRCA1_SGE</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PTEN c.564T&gt;A (p.Y188X) meets criteria to be c...</td>\n",
       "      <td>Matreyek_PTEN_VampSeq</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>PTEN c.964A&gt;T (p.Lys322Ter) meets criteria to ...</td>\n",
       "      <td>Matreyek_PTEN_VampSeq</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>This variant has a BayesDel score &gt; 0.16 and A...</td>\n",
       "      <td>Kato_TP53</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>This variant is absent in the gnomAD cohort (P...</td>\n",
       "      <td>Kato_TP53</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NM_000314.8(PTEN):c.424C&gt;T (p.Arg142Trp) is cu...</td>\n",
       "      <td>Matreyek_PTEN_VampSeq</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>This variant is absent from the gnomAD non-can...</td>\n",
       "      <td>Giacomelli_1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>The NM_000546.6: c.524G&gt;T variant in TP53 is a...</td>\n",
       "      <td>Giacomelli_1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>This variant has a BayesDel score &lt; 0.16 and A...</td>\n",
       "      <td>Kato_TP53</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>This variant has a BayesDel score &lt; 0.16 and A...</td>\n",
       "      <td>Kato_TP53</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>The NM_000546.6 c.892G&gt;T (p.Glu298Ter) is a TP...</td>\n",
       "      <td>Giacomelli_2</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>PTEN c.512dup (p.R172Efs) meets criteria to be...</td>\n",
       "      <td>Matreyek_PTEN_VampSeq</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Summary of interpretation        scoreset_source  \\\n",
       "0    The c.5509T>G variant in BRCA1 is a missense v...      Findlay_BRCA1_SGE   \n",
       "32   PTEN c.564T>A (p.Y188X) meets criteria to be c...  Matreyek_PTEN_VampSeq   \n",
       "52   PTEN c.964A>T (p.Lys322Ter) meets criteria to ...  Matreyek_PTEN_VampSeq   \n",
       "21   This variant has a BayesDel score > 0.16 and A...              Kato_TP53   \n",
       "104  This variant is absent in the gnomAD cohort (P...              Kato_TP53   \n",
       "25   NM_000314.8(PTEN):c.424C>T (p.Arg142Trp) is cu...  Matreyek_PTEN_VampSeq   \n",
       "51   This variant is absent from the gnomAD non-can...           Giacomelli_1   \n",
       "68   The NM_000546.6: c.524G>T variant in TP53 is a...           Giacomelli_1   \n",
       "10   This variant has a BayesDel score < 0.16 and A...              Kato_TP53   \n",
       "52   This variant has a BayesDel score < 0.16 and A...              Kato_TP53   \n",
       "26   The NM_000546.6 c.892G>T (p.Glu298Ter) is a TP...           Giacomelli_2   \n",
       "30   PTEN c.512dup (p.R172Efs) meets criteria to be...  Matreyek_PTEN_VampSeq   \n",
       "\n",
       "     strength  \n",
       "0           8  \n",
       "32          7  \n",
       "52          7  \n",
       "21          6  \n",
       "104         6  \n",
       "25          6  \n",
       "51          6  \n",
       "68          6  \n",
       "10          6  \n",
       "52          6  \n",
       "26          5  \n",
       "30          1  "
      ]
     },
     "execution_count": 298,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax.loc[(dfMax.strength > 0) & (dfMax.loc[:,'Applied Evidence Codes (Not Met)'].str.contains(\"PS3\")),['Summary of interpretation','scoreset_source','strength']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(95, 29)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dfMax[(dfMax.loc[:,'Applied Evidence Codes (Met)'].str.contains(\"BS3\"))].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(74, 29)"
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax[(dfMax.loc[:,'Applied Evidence Codes (Met)'].str.contains(\"BS3\")) & \\\n",
    "     (dfMax.loc[:,'strength'] < 0)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_colwidth', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>avg_score</th>\n",
       "      <th>#Variation</th>\n",
       "      <th>ClinVar Variation Id</th>\n",
       "      <th>Allele Registry Id</th>\n",
       "      <th>HGVS Expressions</th>\n",
       "      <th>HGNC Gene Symbol</th>\n",
       "      <th>Disease</th>\n",
       "      <th>Mondo Id</th>\n",
       "      <th>Mode of Inheritance</th>\n",
       "      <th>Assertion</th>\n",
       "      <th>Applied Evidence Codes (Met)</th>\n",
       "      <th>Applied Evidence Codes (Not Met)</th>\n",
       "      <th>Summary of interpretation</th>\n",
       "      <th>PubMed Articles</th>\n",
       "      <th>Expert Panel</th>\n",
       "      <th>Guideline</th>\n",
       "      <th>Approval Date</th>\n",
       "      <th>Published Date</th>\n",
       "      <th>Retracted</th>\n",
       "      <th>Evidence Repo Link</th>\n",
       "      <th>Uuid</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>metEvidence</th>\n",
       "      <th>Expressions</th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>scoreset_source</th>\n",
       "      <th>strength</th>\n",
       "      <th>strengthAbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>p.Arg267Gln</td>\n",
       "      <td>-2.240292</td>\n",
       "      <td>NM_000546.5(TP53):c.800G&gt;A (p.Arg267Gln)</td>\n",
       "      <td>127823</td>\n",
       "      <td>CA000424</td>\n",
       "      <td>NM_000546.5:c.800G&gt;A, NC_000017.11:g.7673820C&gt;T, CM000679.2:g.7673820C&gt;T, NC_000017.10:g.7577138C&gt;T, CM000679.1:g.7577138C&gt;T, NC_000017.9:g.7517863C&gt;T, NG_017013.2:g.18731G&gt;A, LRG_321:g.18731G&gt;A, LRG_321t1:c.800G&gt;A, NM_001126112.2:c.800G&gt;A, LRG_321t2:c.800G&gt;A, NM_001126113.2:c.800G&gt;A, LRG_321t4:c.800G&gt;A, NM_001126114.2:c.800G&gt;A, LRG_321t3:c.800G&gt;A, NM_001126115.1:c.404G&gt;A, LRG_321t5:c.404G&gt;A, NM_001126116.1:c.404G&gt;A, LRG_321t6:c.404G&gt;A, NM_001126117.1:c.404G&gt;A, LRG_321t7:c.404G&gt;A, NM_001126118.1:c.683G&gt;A, LRG_321t8:c.683G&gt;A, NM_001276695.1:c.683G&gt;A, NM_001276696.1:c.683G&gt;A, NM_001276697.1:c.323G&gt;A, NM_001276698.1:c.323G&gt;A, NM_001276699.1:c.323G&gt;A, NM_001276760.1:c.683G&gt;A, NM_001276761.1:c.683G&gt;A, NM_001276695.2:c.683G&gt;A, NM_001276696.2:c.683G&gt;A, NM_001276697.2:c.323G&gt;A, NM_001276698.2:c.323G&gt;A, NM_001276699.2:c.323G&gt;A, NM_001276760.2:c.683G&gt;A, NM_001276761.2:c.683G&gt;A, ENST00000269305.8:c.800G&gt;A, ENST00000359597.8:n.800G&gt;A, ENST00000413465.6:n.782+361G&gt;A, ENST00000420246.6:c.800G&gt;A, ENST00000445888.6:c.800G&gt;A, ENST00000455263.6:c.800G&gt;A, ENST00000504290.5:c.404G&gt;A, ENST00000504937.5:c.404G&gt;A, ENST00000509690.5:c.404G&gt;A, ENST00000510385.5:c.404G&gt;A, ENST00000610292.4:c.683G&gt;A, ENST00000610538.4:c.683G&gt;A, ENST00000610623.4:c.323G&gt;A, ENST00000615910.4:n.767G&gt;A, ENST00000617185.4:c.800G&gt;A, ENST00000618944.4:c.323G&gt;A, ENST00000619186.4:c.323G&gt;A, ENST00000619485.4:c.683G&gt;A, ENST00000620739.4:c.683G&gt;A, ENST00000622645.4:c.683G&gt;A, ENST00000635293.1:c.683G&gt;A, NM_000546.5(TP53):c.800G&gt;A (p.Arg267Gln)</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Li-Fraumeni syndrome</td>\n",
       "      <td>MONDO:0018875</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>PP3, PS4_Supporting, BS3_Supporting</td>\n",
       "      <td>PM1, PM5, BA1, BP4, BS1</td>\n",
       "      <td>This variant has a BayesDel score &gt; 0.16 and Align GVGD (Zebrafish) is Class 15 or higher (PP3). This variant has been reported in 2 probands meeting Chrompret criteria (PS4_Supporting; PMID:25584008, internal laboratory contributor). Transactivation assays show partially functional variant according to Kato, et al. and there is no evidence of a dominant negative effect or loss of function according to Giacomelli, et al. (BS3_Supporting; PMID: 12826609, 30224644). In summary, the clinical significance of TP53 c.800G&gt;A (p.Arg267Gln) is uncertain for Li-Fraumeni syndrome. ACMG/AMP criteria applied, as specified by the TP53 Expert Panel: PP3, PS4_Supporting, BS3_Supporting.</td>\n",
       "      <td>1562462, 25584008, 27210295, 10435620, 21343334, 25584008, 30224644, 12826609, 29979965</td>\n",
       "      <td>TP53 VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020-08-11</td>\n",
       "      <td>2020-08-14</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classification/CA000424/MONDO:0018875/009</td>\n",
       "      <td>ca16e21b-853c-433e-88e5-601b2de6c525</td>\n",
       "      <td>TP53</td>\n",
       "      <td>[PP3, PS4_Supporting, BS3_Supporting]</td>\n",
       "      <td>(NM_000546.5:c.800G&gt;A, NC_000017.11:g.7673820C&gt;T, CM000679.2:g.7673820C&gt;T, NC_000017.10:g.7577138C&gt;T, CM000679.1:g.7577138C&gt;T, NC_000017.9:g.7517863C&gt;T, NG_017013.2:g.18731G&gt;A, LRG_321:g.18731G&gt;A, LRG_321t1:c.800G&gt;A, NM_001126112.2:c.800G&gt;A, LRG_321t2:c.800G&gt;A, NM_001126113.2:c.800G&gt;A, LRG_321t4:c.800G&gt;A, NM_001126114.2:c.800G&gt;A, LRG_321t3:c.800G&gt;A, NM_001126115.1:c.404G&gt;A, LRG_321t5:c.404G&gt;A, NM_001126116.1:c.404G&gt;A, LRG_321t6:c.404G&gt;A, NM_001126117.1:c.404G&gt;A, LRG_321t7:c.404G&gt;A, NM_001126118.1:c.683G&gt;A, LRG_321t8:c.683G&gt;A, NM_001276695.1:c.683G&gt;A, NM_001276696.1:c.683G&gt;A, NM_001276697.1:c.323G&gt;A, NM_001276698.1:c.323G&gt;A, NM_001276699.1:c.323G&gt;A, NM_001276760.1:c.683G&gt;A, NM_001276761.1:c.683G&gt;A, NM_001276695.2:c.683G&gt;A, NM_001276696.2:c.683G&gt;A, NM_001276697.2:c.323G&gt;A, NM_001276698.2:c.323G&gt;A, NM_001276699.2:c.323G&gt;A, NM_001276760.2:c.683G&gt;A, NM_001276761.2:c.683G&gt;A, ENST00000269305.8:c.800G&gt;A, ENST00000359597.8:n.800G&gt;A, ENST00000413465.6:n.782+361G&gt;A, ENST00000420246.6:c.800G&gt;A, ENST00000445888.6:c.800G&gt;A, ENST00000455263.6:c.800G&gt;A, ENST00000504290.5:c.404G&gt;A, ENST00000504937.5:c.404G&gt;A, ENST00000509690.5:c.404G&gt;A, ENST00000510385.5:c.404G&gt;A, ENST00000610292.4:c.683G&gt;A, ENST00000610538.4:c.683G&gt;A, ENST00000610623.4:c.323G&gt;A, ENST00000615910.4:n.767G&gt;A, ENST00000617185.4:c.800G&gt;A, ENST00000618944.4:c.323G&gt;A, ENST00000619186.4:c.323G&gt;A, ENST00000619485.4:c.683G&gt;A, ENST00000620739.4:c.683G&gt;A, ENST00000622645.4:c.683G&gt;A, ENST00000635293.1:c.683G&gt;A, NM_000546.5(TP53):c.800G&gt;A (p.Arg267Gln))</td>\n",
       "      <td>p.Arg267Gln</td>\n",
       "      <td>Kato_TP53</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>p.Arg175Leu</td>\n",
       "      <td>0.940932</td>\n",
       "      <td>NM_000546.6(TP53):c.524G&gt;T (p.Arg175Leu)</td>\n",
       "      <td>182963</td>\n",
       "      <td>CA000252</td>\n",
       "      <td>NM_000546.6:c.524G&gt;T, NC_000017.11:g.7675088C&gt;A, CM000679.2:g.7675088C&gt;A, NC_000017.10:g.7578406C&gt;A, CM000679.1:g.7578406C&gt;A, NC_000017.9:g.7519131C&gt;A, NG_017013.2:g.17463G&gt;T, LRG_321:g.17463G&gt;T, ENST00000503591.2:c.524G&gt;T, ENST00000508793.6:c.524G&gt;T, ENST00000509690.6:c.128G&gt;T, ENST00000514944.6:c.245G&gt;T, ENST00000604348.6:c.503G&gt;T, ENST00000269305.9:c.524G&gt;T, ENST00000269305.8:c.524G&gt;T, ENST00000359597.8:c.524G&gt;T, ENST00000413465.6:c.524G&gt;T, ENST00000420246.6:c.524G&gt;T, ENST00000445888.6:c.524G&gt;T, ENST00000455263.6:c.524G&gt;T, ENST00000504290.5:c.128G&gt;T, ENST00000504937.5:c.128G&gt;T, ENST00000505014.5:n.780G&gt;T, ENST00000509690.5:c.128G&gt;T, ENST00000510385.5:c.128G&gt;T, ENST00000514944.5:c.245G&gt;T, ENST00000574684.1:n.32G&gt;T, ENST00000610292.4:c.407G&gt;T, ENST00000610538.4:c.407G&gt;T, ENST00000610623.4:c.47G&gt;T, ENST00000615910.4:c.491G&gt;T, ENST00000617185.4:c.524G&gt;T, ENST00000618944.4:c.47G&gt;T, ENST00000619186.4:c.47G&gt;T, ENST00000619485.4:c.407G&gt;T, ENST00000620739.4:c.407G&gt;T, ENST00000622645.4:c.407G&gt;T, ENST00000635293.1:c.407G&gt;T, NM_000546.5:c.524G&gt;T, LRG_321t1:c.524G&gt;T, NM_001126112.2:c.524G&gt;T, LRG_321t2:c.524G&gt;T, NM_001126113.2:c.524G&gt;T, LRG_321t4:c.524G&gt;T, NM_001126114.2:c.524G&gt;T, LRG_321t3:c.524G&gt;T, NM_001126115.1:c.128G&gt;T, LRG_321t5:c.128G&gt;T, NM_001126116.1:c.128G&gt;T, LRG_321t6:c.128G&gt;T, NM_001126117.1:c.128G&gt;T, LRG_321t7:c.128G&gt;T, NM_001126118.1:c.407G&gt;T, LRG_321t8:c.407G&gt;T, NM_001276695.1:c.407G&gt;T, NM_001276696.1:c.407G&gt;T, NM_001276697.1:c.47G&gt;T, NM_001276698.1:c.47G&gt;T, NM_001276699.1:c.47G&gt;T, NM_001276760.1:c.407G&gt;T, NM_001276761.1:c.407G&gt;T, NM_001276695.2:c.407G&gt;T, NM_001276696.2:c.407G&gt;T, NM_001276697.2:c.47G&gt;T, NM_001276698.2:c.47G&gt;T, NM_001276699.2:c.47G&gt;T, NM_001276760.2:c.407G&gt;T, NM_001276761.2:c.407G&gt;T, NM_001126112.3:c.524G&gt;T, NM_001126113.3:c.524G&gt;T, NM_001126114.3:c.524G&gt;T, NM_001126115.2:c.128G&gt;T, NM_001126116.2:c.128G&gt;T, NM_001126117.2:c.128G&gt;T, NM_001126118.2:c.407G&gt;T, NM_001276695.3:c.407G&gt;T, NM_001276696.3:c.407G&gt;T, NM_001276697.3:c.47G&gt;T, NM_001276698.3:c.47G&gt;T, NM_001276699.3:c.47G&gt;T, NM_001276760.3:c.407G&gt;T, NM_001276761.3:c.407G&gt;T, NM_000546.6(TP53):c.524G&gt;T (p.Arg175Leu)</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Li-Fraumeni syndrome</td>\n",
       "      <td>MONDO:0018875</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PM1, PM5, PP4, PM2_Supporting, BS3_Supporting, PS4_Moderate, PP3_Moderate</td>\n",
       "      <td>PP1, BS2, BA1, BP4, PS3, PS2, PS1, BS1</td>\n",
       "      <td>The NM_000546.6: c.524G&gt;T variant in TP53 is a missense variant predicted to cause substitution of arginine by leucine at amino acid 175 (p.Arg175Leu). This variant is absent from gnomAD v4.1.0 (PM2_Supporting). This variant has been reported in 4 unrelated probands meeting Revised Chompret criteria. Based on this evidence, this variant scores 2 total points meeting the TP53 VCEP phenotype scoring criteria of 2-3.5 points. (PS4_Moderate; PMID: 16707427; Internal lab contributors: Invitae, Ambry). In vitro assays performed in yeast and/or human cell lines showed partially functional transactivation, and retained growth suppression activity indicating that this variant does not impact protein function (BS3_Supporting; PMIDs: 12826609, 29979965, 30224644).Computational predictor scores (BayesDel = 0.57303; Align GVGD = Class C65) are above recommended thresholds (BayesDel &gt; 0.16 and an Align GVGD Class of 65), evidence that correlates with impact to TP53 via protein change (PP3_Moderate). At least one individual with this variant was found to have a variant allele fraction of ≤35%, which is a significant predictor of variant pathogenicity (PP4_Moderate, PMID: 34906512, Internal lab contributor: Invitae). This variant resides within a codon (NM_00546.4: 175, 245, 248, 249, 273, 282) of TP53 that is defined as a mutational hotspot by the ClinGen TP53 VCEP (PM1; PMID: 8023157). Another missense variant(c.524G&gt;A, p.Arg175His) (ClinVar Variation ID: 12374), in the same codon has been classified as pathogenic for Li-Fraumeni syndrome by the ClinGen TP53 VCEP’s specifications. (PM5). In summary, this variant meets the criteria to be classified as Likely Pathogenic for Li Fraumeni syndrome based on the ACMG/AMP criteria applied, as specified by the ClinGen TP53 VCEP: PS4_Moderate, PM2_Supporting, BS3_Supporting, PP3_Moderate, PP4, PM1, PM5 (Bayesian Points: 9; VCEP specifications version 2.0; 9/6/2024)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53 VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-09-06</td>\n",
       "      <td>2024-09-06</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classification/CA000252/MONDO:0018875/009</td>\n",
       "      <td>6aa7e5db-edd6-4608-a046-e82aa9ea21df</td>\n",
       "      <td>TP53</td>\n",
       "      <td>[PM1, PM5, PP4, PM2_Supporting, BS3_Supporting, PS4_Moderate, PP3_Moderate]</td>\n",
       "      <td>(NM_000546.6:c.524G&gt;T, NC_000017.11:g.7675088C&gt;A, CM000679.2:g.7675088C&gt;A, NC_000017.10:g.7578406C&gt;A, CM000679.1:g.7578406C&gt;A, NC_000017.9:g.7519131C&gt;A, NG_017013.2:g.17463G&gt;T, LRG_321:g.17463G&gt;T, ENST00000503591.2:c.524G&gt;T, ENST00000508793.6:c.524G&gt;T, ENST00000509690.6:c.128G&gt;T, ENST00000514944.6:c.245G&gt;T, ENST00000604348.6:c.503G&gt;T, ENST00000269305.9:c.524G&gt;T, ENST00000269305.8:c.524G&gt;T, ENST00000359597.8:c.524G&gt;T, ENST00000413465.6:c.524G&gt;T, ENST00000420246.6:c.524G&gt;T, ENST00000445888.6:c.524G&gt;T, ENST00000455263.6:c.524G&gt;T, ENST00000504290.5:c.128G&gt;T, ENST00000504937.5:c.128G&gt;T, ENST00000505014.5:n.780G&gt;T, ENST00000509690.5:c.128G&gt;T, ENST00000510385.5:c.128G&gt;T, ENST00000514944.5:c.245G&gt;T, ENST00000574684.1:n.32G&gt;T, ENST00000610292.4:c.407G&gt;T, ENST00000610538.4:c.407G&gt;T, ENST00000610623.4:c.47G&gt;T, ENST00000615910.4:c.491G&gt;T, ENST00000617185.4:c.524G&gt;T, ENST00000618944.4:c.47G&gt;T, ENST00000619186.4:c.47G&gt;T, ENST00000619485.4:c.407G&gt;T, ENST00000620739.4:c.407G&gt;T, ENST00000622645.4:c.407G&gt;T, ENST00000635293.1:c.407G&gt;T, NM_000546.5:c.524G&gt;T, LRG_321t1:c.524G&gt;T, NM_001126112.2:c.524G&gt;T, LRG_321t2:c.524G&gt;T, NM_001126113.2:c.524G&gt;T, LRG_321t4:c.524G&gt;T, NM_001126114.2:c.524G&gt;T, LRG_321t3:c.524G&gt;T, NM_001126115.1:c.128G&gt;T, LRG_321t5:c.128G&gt;T, NM_001126116.1:c.128G&gt;T, LRG_321t6:c.128G&gt;T, NM_001126117.1:c.128G&gt;T, LRG_321t7:c.128G&gt;T, NM_001126118.1:c.407G&gt;T, LRG_321t8:c.407G&gt;T, NM_001276695.1:c.407G&gt;T, NM_001276696.1:c.407G&gt;T, NM_001276697.1:c.47G&gt;T, NM_001276698.1:c.47G&gt;T, NM_001276699.1:c.47G&gt;T, NM_001276760.1:c.407G&gt;T, NM_001276761.1:c.407G&gt;T, NM_001276695.2:c.407G&gt;T, NM_001276696.2:c.407G&gt;T, NM_001276697.2:c.47G&gt;T, NM_001276698.2:c.47G&gt;T, NM_001276699.2:c.47G&gt;T, NM_001276760.2:c.407G&gt;T, NM_001276761.2:c.407G&gt;T, NM_001126112.3:c.524G&gt;T, NM_001126113.3:c.524G&gt;T, NM_001126114.3:c.524G&gt;T, NM_001126115.2:c.128G&gt;T, NM_001126116.2:c.128G&gt;T, NM_001126117.2:c.128G&gt;T, NM_001126118.2:c.407G&gt;T, NM_001276695.3:c.407G&gt;T, NM_001276696.3:c.407G&gt;T, NM_001276697.3:c.47G&gt;T, NM_001276698.3:c.47G&gt;T, NM_001276699.3:c.47G&gt;T, NM_001276760.3:c.407G&gt;T, NM_001276761.3:c.407G&gt;T, NM_000546.6(TP53):c.524G&gt;T (p.Arg175Leu))</td>\n",
       "      <td>p.Arg175Leu</td>\n",
       "      <td>Giacomelli_1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>p.Ala102Gly</td>\n",
       "      <td>-0.046066</td>\n",
       "      <td>NM_007294.4(BRCA1):c.305C&gt;G (p.Ala102Gly)</td>\n",
       "      <td>37505</td>\n",
       "      <td>CA002009</td>\n",
       "      <td>NM_007294.4:c.305C&gt;G, NC_000017.11:g.43104258G&gt;C, CM000679.2:g.43104258G&gt;C, NC_000017.10:g.41256275G&gt;C, CM000679.1:g.41256275G&gt;C, NC_000017.9:g.38509801G&gt;C, NG_005905.2:g.113726C&gt;G, LRG_292:g.113726C&gt;G, ENST00000354071.8:n.369C&gt;G, ENST00000461574.2:c.305C&gt;G, ENST00000470026.6:c.305C&gt;G, ENST00000473961.6:c.305C&gt;G, ENST00000476777.6:c.305C&gt;G, ENST00000477152.6:c.227C&gt;G, ENST00000478531.6:c.305C&gt;G, ENST00000489037.2:c.227C&gt;G, ENST00000493919.6:c.164C&gt;G, ENST00000494123.6:c.305C&gt;G, ENST00000497488.2:c.-218-9398C&gt;G, ENST00000618469.2:c.305C&gt;G, ENST00000634433.2:c.305C&gt;G, ENST00000644379.2:c.305C&gt;G, ENST00000644555.2:c.164C&gt;G, ENST00000652672.2:c.164C&gt;G, ENST00000484087.6:c.305C&gt;G, ENST00000700083.1:n.1276C&gt;G, ENST00000700182.1:c.227C&gt;G, ENST00000700183.1:c.*219C&gt;G, ENST00000700184.1:n.548C&gt;G, ENST00000357654.9:c.305C&gt;G, ENST00000471181.7:c.305C&gt;G, ENST00000642945.1:c.*179C&gt;G, ENST00000644555.1:c.164C&gt;G, ENST00000652672.1:c.164C&gt;G, ENST00000352993.7:c.305C&gt;G, ENST00000354071.7:c.305C&gt;G, ENST00000357654.7:c.305C&gt;G, ENST00000461221.5:c.*91C&gt;G, ENST00000461798.5:c.*91C&gt;G, ENST00000468300.5:c.305C&gt;G, ENST00000470026.5:c.305C&gt;G, ENST00000471181.6:c.305C&gt;G, ENST00000473961.5:c.28C&gt;G, ENST00000476777.5:c.305C&gt;G, ENST00000477152.5:c.227C&gt;G, ENST00000478531.5:c.305C&gt;G, ENST00000484087.5:c.53C&gt;G, ENST00000487825.5:c.53C&gt;G, ENST00000489037.1:c.227C&gt;G, ENST00000491747.6:c.305C&gt;G, ENST00000492859.5:c.*241C&gt;G, ENST00000493795.5:c.164C&gt;G, ENST00000493919.5:c.164C&gt;G, ENST00000494123.5:c.305C&gt;G, ENST00000497488.1:c.-218-9398C&gt;G, ENST00000586385.5:c.4+20924C&gt;G, ENST00000591534.5:c.-44+21013C&gt;G, ENST00000591849.5:c.-99+21013C&gt;G, ENST00000634433.1:c.305C&gt;G, NM_007294.3:c.305C&gt;G, LRG_292t1:c.305C&gt;G, NM_007297.3:c.164C&gt;G, NM_007298.3:c.305C&gt;G, NM_007299.3:c.305C&gt;G, NM_007300.3:c.305C&gt;G, NR_027676.1:n.444C&gt;G, NM_007297.4:c.164C&gt;G, NM_007299.4:c.305C&gt;G, NM_007300.4:c.305C&gt;G, NR_027676.2:n.485C&gt;G, NM_007294.4(BRCA1):c.305C&gt;G (p.Ala102Gly)</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>BRCA1-related cancer predisposition</td>\n",
       "      <td>MONDO:0011450</td>\n",
       "      <td>Autosomal dominant inheritance</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PM2_Supporting, BP1_Strong, BS3, BP5_Strong</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The c.305C&gt;G variant in BRCA1 is a missense variant predicted to cause substitution of Alanine by Glycine at amino acid 102 (p.Ala102Gly). This variant is absent from gnomAD v2.1 (exomes only, non-cancer subset, read depth ≥25) and gnomAD v3.1 (non-cancer subset, read depth ≥25) (PM2_Supporting met). This missense variant is located outside of a key functional domain and was not predicted to alter mRNA splicing using the SpliceAI predictor (score 0.00, score threshold &lt;0.1) (BP1_Strong met). Reported by one calibrated study to exhibit protein function similar to benign control variants (PMID: 30219179) (BS3 met). Multifactorial likelihood ratio analysis using clinically calibrated data produced a combined LR for this variant of 0.015 (based on Co-occurrence LR=1.177; Family History LR=0.0127), below the threshold for strong benign evidence (LR &lt;0.05) (BP5_Strong met; PMID: 31131967, 31853058). In summary, this variant meets the criteria to be classified as a Benign variant for BRCA1-related cancer predisposition based on the ACMG/AMP criteria applied as specified by the ENIGMA BRCA1/2 VCEP (PM2_Supporting, BP1_Strong, BS3, BP5_Strong).</td>\n",
       "      <td>30219179</td>\n",
       "      <td>ENIGMA BRCA1 and BRCA2 VCEP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-12</td>\n",
       "      <td>2024-06-11</td>\n",
       "      <td>False</td>\n",
       "      <td>https://erepo.genome.network/evrepo/ui/classification/CA002009/MONDO:0011450/092</td>\n",
       "      <td>de5081dd-8e3e-444d-9826-44632cbee0f3</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>[PM2_Supporting, BP1_Strong, BS3, BP5_Strong]</td>\n",
       "      <td>(NM_007294.4:c.305C&gt;G, NC_000017.11:g.43104258G&gt;C, CM000679.2:g.43104258G&gt;C, NC_000017.10:g.41256275G&gt;C, CM000679.1:g.41256275G&gt;C, NC_000017.9:g.38509801G&gt;C, NG_005905.2:g.113726C&gt;G, LRG_292:g.113726C&gt;G, ENST00000354071.8:n.369C&gt;G, ENST00000461574.2:c.305C&gt;G, ENST00000470026.6:c.305C&gt;G, ENST00000473961.6:c.305C&gt;G, ENST00000476777.6:c.305C&gt;G, ENST00000477152.6:c.227C&gt;G, ENST00000478531.6:c.305C&gt;G, ENST00000489037.2:c.227C&gt;G, ENST00000493919.6:c.164C&gt;G, ENST00000494123.6:c.305C&gt;G, ENST00000497488.2:c.-218-9398C&gt;G, ENST00000618469.2:c.305C&gt;G, ENST00000634433.2:c.305C&gt;G, ENST00000644379.2:c.305C&gt;G, ENST00000644555.2:c.164C&gt;G, ENST00000652672.2:c.164C&gt;G, ENST00000484087.6:c.305C&gt;G, ENST00000700083.1:n.1276C&gt;G, ENST00000700182.1:c.227C&gt;G, ENST00000700183.1:c.*219C&gt;G, ENST00000700184.1:n.548C&gt;G, ENST00000357654.9:c.305C&gt;G, ENST00000471181.7:c.305C&gt;G, ENST00000642945.1:c.*179C&gt;G, ENST00000644555.1:c.164C&gt;G, ENST00000652672.1:c.164C&gt;G, ENST00000352993.7:c.305C&gt;G, ENST00000354071.7:c.305C&gt;G, ENST00000357654.7:c.305C&gt;G, ENST00000461221.5:c.*91C&gt;G, ENST00000461798.5:c.*91C&gt;G, ENST00000468300.5:c.305C&gt;G, ENST00000470026.5:c.305C&gt;G, ENST00000471181.6:c.305C&gt;G, ENST00000473961.5:c.28C&gt;G, ENST00000476777.5:c.305C&gt;G, ENST00000477152.5:c.227C&gt;G, ENST00000478531.5:c.305C&gt;G, ENST00000484087.5:c.53C&gt;G, ENST00000487825.5:c.53C&gt;G, ENST00000489037.1:c.227C&gt;G, ENST00000491747.6:c.305C&gt;G, ENST00000492859.5:c.*241C&gt;G, ENST00000493795.5:c.164C&gt;G, ENST00000493919.5:c.164C&gt;G, ENST00000494123.5:c.305C&gt;G, ENST00000497488.1:c.-218-9398C&gt;G, ENST00000586385.5:c.4+20924C&gt;G, ENST00000591534.5:c.-44+21013C&gt;G, ENST00000591849.5:c.-99+21013C&gt;G, ENST00000634433.1:c.305C&gt;G, NM_007294.3:c.305C&gt;G, LRG_292t1:c.305C&gt;G, NM_007297.3:c.164C&gt;G, NM_007298.3:c.305C&gt;G, NM_007299.3:c.305C&gt;G, NM_007300.3:c.305C&gt;G, NR_027676.1:n.444C&gt;G, NM_007297.4:c.164C&gt;G, NM_007299.4:c.305C&gt;G, NM_007300.4:c.305C&gt;G, NR_027676.2:n.485C&gt;G, NM_007294.4(BRCA1):c.305C&gt;G (p.Ala102Gly))</td>\n",
       "      <td>p.Ala102Gly</td>\n",
       "      <td>Starita_BRCA1_HDR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hgvs_pro  avg_score                                 #Variation  \\\n",
       "72  p.Arg267Gln  -2.240292   NM_000546.5(TP53):c.800G>A (p.Arg267Gln)   \n",
       "68  p.Arg175Leu   0.940932   NM_000546.6(TP53):c.524G>T (p.Arg175Leu)   \n",
       "0   p.Ala102Gly  -0.046066  NM_007294.4(BRCA1):c.305C>G (p.Ala102Gly)   \n",
       "\n",
       "   ClinVar Variation Id Allele Registry Id  \\\n",
       "72               127823           CA000424   \n",
       "68               182963           CA000252   \n",
       "0                 37505           CA002009   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            HGVS Expressions  \\\n",
       "72                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 NM_000546.5:c.800G>A, NC_000017.11:g.7673820C>T, CM000679.2:g.7673820C>T, NC_000017.10:g.7577138C>T, CM000679.1:g.7577138C>T, NC_000017.9:g.7517863C>T, NG_017013.2:g.18731G>A, LRG_321:g.18731G>A, LRG_321t1:c.800G>A, NM_001126112.2:c.800G>A, LRG_321t2:c.800G>A, NM_001126113.2:c.800G>A, LRG_321t4:c.800G>A, NM_001126114.2:c.800G>A, LRG_321t3:c.800G>A, NM_001126115.1:c.404G>A, LRG_321t5:c.404G>A, NM_001126116.1:c.404G>A, LRG_321t6:c.404G>A, NM_001126117.1:c.404G>A, LRG_321t7:c.404G>A, NM_001126118.1:c.683G>A, LRG_321t8:c.683G>A, NM_001276695.1:c.683G>A, NM_001276696.1:c.683G>A, NM_001276697.1:c.323G>A, NM_001276698.1:c.323G>A, NM_001276699.1:c.323G>A, NM_001276760.1:c.683G>A, NM_001276761.1:c.683G>A, NM_001276695.2:c.683G>A, NM_001276696.2:c.683G>A, NM_001276697.2:c.323G>A, NM_001276698.2:c.323G>A, NM_001276699.2:c.323G>A, NM_001276760.2:c.683G>A, NM_001276761.2:c.683G>A, ENST00000269305.8:c.800G>A, ENST00000359597.8:n.800G>A, ENST00000413465.6:n.782+361G>A, ENST00000420246.6:c.800G>A, ENST00000445888.6:c.800G>A, ENST00000455263.6:c.800G>A, ENST00000504290.5:c.404G>A, ENST00000504937.5:c.404G>A, ENST00000509690.5:c.404G>A, ENST00000510385.5:c.404G>A, ENST00000610292.4:c.683G>A, ENST00000610538.4:c.683G>A, ENST00000610623.4:c.323G>A, ENST00000615910.4:n.767G>A, ENST00000617185.4:c.800G>A, ENST00000618944.4:c.323G>A, ENST00000619186.4:c.323G>A, ENST00000619485.4:c.683G>A, ENST00000620739.4:c.683G>A, ENST00000622645.4:c.683G>A, ENST00000635293.1:c.683G>A, NM_000546.5(TP53):c.800G>A (p.Arg267Gln)   \n",
       "68  NM_000546.6:c.524G>T, NC_000017.11:g.7675088C>A, CM000679.2:g.7675088C>A, NC_000017.10:g.7578406C>A, CM000679.1:g.7578406C>A, NC_000017.9:g.7519131C>A, NG_017013.2:g.17463G>T, LRG_321:g.17463G>T, ENST00000503591.2:c.524G>T, ENST00000508793.6:c.524G>T, ENST00000509690.6:c.128G>T, ENST00000514944.6:c.245G>T, ENST00000604348.6:c.503G>T, ENST00000269305.9:c.524G>T, ENST00000269305.8:c.524G>T, ENST00000359597.8:c.524G>T, ENST00000413465.6:c.524G>T, ENST00000420246.6:c.524G>T, ENST00000445888.6:c.524G>T, ENST00000455263.6:c.524G>T, ENST00000504290.5:c.128G>T, ENST00000504937.5:c.128G>T, ENST00000505014.5:n.780G>T, ENST00000509690.5:c.128G>T, ENST00000510385.5:c.128G>T, ENST00000514944.5:c.245G>T, ENST00000574684.1:n.32G>T, ENST00000610292.4:c.407G>T, ENST00000610538.4:c.407G>T, ENST00000610623.4:c.47G>T, ENST00000615910.4:c.491G>T, ENST00000617185.4:c.524G>T, ENST00000618944.4:c.47G>T, ENST00000619186.4:c.47G>T, ENST00000619485.4:c.407G>T, ENST00000620739.4:c.407G>T, ENST00000622645.4:c.407G>T, ENST00000635293.1:c.407G>T, NM_000546.5:c.524G>T, LRG_321t1:c.524G>T, NM_001126112.2:c.524G>T, LRG_321t2:c.524G>T, NM_001126113.2:c.524G>T, LRG_321t4:c.524G>T, NM_001126114.2:c.524G>T, LRG_321t3:c.524G>T, NM_001126115.1:c.128G>T, LRG_321t5:c.128G>T, NM_001126116.1:c.128G>T, LRG_321t6:c.128G>T, NM_001126117.1:c.128G>T, LRG_321t7:c.128G>T, NM_001126118.1:c.407G>T, LRG_321t8:c.407G>T, NM_001276695.1:c.407G>T, NM_001276696.1:c.407G>T, NM_001276697.1:c.47G>T, NM_001276698.1:c.47G>T, NM_001276699.1:c.47G>T, NM_001276760.1:c.407G>T, NM_001276761.1:c.407G>T, NM_001276695.2:c.407G>T, NM_001276696.2:c.407G>T, NM_001276697.2:c.47G>T, NM_001276698.2:c.47G>T, NM_001276699.2:c.47G>T, NM_001276760.2:c.407G>T, NM_001276761.2:c.407G>T, NM_001126112.3:c.524G>T, NM_001126113.3:c.524G>T, NM_001126114.3:c.524G>T, NM_001126115.2:c.128G>T, NM_001126116.2:c.128G>T, NM_001126117.2:c.128G>T, NM_001126118.2:c.407G>T, NM_001276695.3:c.407G>T, NM_001276696.3:c.407G>T, NM_001276697.3:c.47G>T, NM_001276698.3:c.47G>T, NM_001276699.3:c.47G>T, NM_001276760.3:c.407G>T, NM_001276761.3:c.407G>T, NM_000546.6(TP53):c.524G>T (p.Arg175Leu)   \n",
       "0                                                                                                                                                                                      NM_007294.4:c.305C>G, NC_000017.11:g.43104258G>C, CM000679.2:g.43104258G>C, NC_000017.10:g.41256275G>C, CM000679.1:g.41256275G>C, NC_000017.9:g.38509801G>C, NG_005905.2:g.113726C>G, LRG_292:g.113726C>G, ENST00000354071.8:n.369C>G, ENST00000461574.2:c.305C>G, ENST00000470026.6:c.305C>G, ENST00000473961.6:c.305C>G, ENST00000476777.6:c.305C>G, ENST00000477152.6:c.227C>G, ENST00000478531.6:c.305C>G, ENST00000489037.2:c.227C>G, ENST00000493919.6:c.164C>G, ENST00000494123.6:c.305C>G, ENST00000497488.2:c.-218-9398C>G, ENST00000618469.2:c.305C>G, ENST00000634433.2:c.305C>G, ENST00000644379.2:c.305C>G, ENST00000644555.2:c.164C>G, ENST00000652672.2:c.164C>G, ENST00000484087.6:c.305C>G, ENST00000700083.1:n.1276C>G, ENST00000700182.1:c.227C>G, ENST00000700183.1:c.*219C>G, ENST00000700184.1:n.548C>G, ENST00000357654.9:c.305C>G, ENST00000471181.7:c.305C>G, ENST00000642945.1:c.*179C>G, ENST00000644555.1:c.164C>G, ENST00000652672.1:c.164C>G, ENST00000352993.7:c.305C>G, ENST00000354071.7:c.305C>G, ENST00000357654.7:c.305C>G, ENST00000461221.5:c.*91C>G, ENST00000461798.5:c.*91C>G, ENST00000468300.5:c.305C>G, ENST00000470026.5:c.305C>G, ENST00000471181.6:c.305C>G, ENST00000473961.5:c.28C>G, ENST00000476777.5:c.305C>G, ENST00000477152.5:c.227C>G, ENST00000478531.5:c.305C>G, ENST00000484087.5:c.53C>G, ENST00000487825.5:c.53C>G, ENST00000489037.1:c.227C>G, ENST00000491747.6:c.305C>G, ENST00000492859.5:c.*241C>G, ENST00000493795.5:c.164C>G, ENST00000493919.5:c.164C>G, ENST00000494123.5:c.305C>G, ENST00000497488.1:c.-218-9398C>G, ENST00000586385.5:c.4+20924C>G, ENST00000591534.5:c.-44+21013C>G, ENST00000591849.5:c.-99+21013C>G, ENST00000634433.1:c.305C>G, NM_007294.3:c.305C>G, LRG_292t1:c.305C>G, NM_007297.3:c.164C>G, NM_007298.3:c.305C>G, NM_007299.3:c.305C>G, NM_007300.3:c.305C>G, NR_027676.1:n.444C>G, NM_007297.4:c.164C>G, NM_007299.4:c.305C>G, NM_007300.4:c.305C>G, NR_027676.2:n.485C>G, NM_007294.4(BRCA1):c.305C>G (p.Ala102Gly)   \n",
       "\n",
       "   HGNC Gene Symbol                              Disease       Mondo Id  \\\n",
       "72             TP53                 Li-Fraumeni syndrome  MONDO:0018875   \n",
       "68             TP53                 Li-Fraumeni syndrome  MONDO:0018875   \n",
       "0             BRCA1  BRCA1-related cancer predisposition  MONDO:0011450   \n",
       "\n",
       "               Mode of Inheritance               Assertion  \\\n",
       "72  Autosomal dominant inheritance  Uncertain Significance   \n",
       "68  Autosomal dominant inheritance       Likely Pathogenic   \n",
       "0   Autosomal dominant inheritance                  Benign   \n",
       "\n",
       "                                                 Applied Evidence Codes (Met)  \\\n",
       "72                                        PP3, PS4_Supporting, BS3_Supporting   \n",
       "68  PM1, PM5, PP4, PM2_Supporting, BS3_Supporting, PS4_Moderate, PP3_Moderate   \n",
       "0                                 PM2_Supporting, BP1_Strong, BS3, BP5_Strong   \n",
       "\n",
       "          Applied Evidence Codes (Not Met)  \\\n",
       "72                 PM1, PM5, BA1, BP4, BS1   \n",
       "68  PP1, BS2, BA1, BP4, PS3, PS2, PS1, BS1   \n",
       "0                                      NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Summary of interpretation  \\\n",
       "72                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               This variant has a BayesDel score > 0.16 and Align GVGD (Zebrafish) is Class 15 or higher (PP3). This variant has been reported in 2 probands meeting Chrompret criteria (PS4_Supporting; PMID:25584008, internal laboratory contributor). Transactivation assays show partially functional variant according to Kato, et al. and there is no evidence of a dominant negative effect or loss of function according to Giacomelli, et al. (BS3_Supporting; PMID: 12826609, 30224644). In summary, the clinical significance of TP53 c.800G>A (p.Arg267Gln) is uncertain for Li-Fraumeni syndrome. ACMG/AMP criteria applied, as specified by the TP53 Expert Panel: PP3, PS4_Supporting, BS3_Supporting.   \n",
       "68  The NM_000546.6: c.524G>T variant in TP53 is a missense variant predicted to cause substitution of arginine by leucine at amino acid 175 (p.Arg175Leu). This variant is absent from gnomAD v4.1.0 (PM2_Supporting). This variant has been reported in 4 unrelated probands meeting Revised Chompret criteria. Based on this evidence, this variant scores 2 total points meeting the TP53 VCEP phenotype scoring criteria of 2-3.5 points. (PS4_Moderate; PMID: 16707427; Internal lab contributors: Invitae, Ambry). In vitro assays performed in yeast and/or human cell lines showed partially functional transactivation, and retained growth suppression activity indicating that this variant does not impact protein function (BS3_Supporting; PMIDs: 12826609, 29979965, 30224644).Computational predictor scores (BayesDel = 0.57303; Align GVGD = Class C65) are above recommended thresholds (BayesDel > 0.16 and an Align GVGD Class of 65), evidence that correlates with impact to TP53 via protein change (PP3_Moderate). At least one individual with this variant was found to have a variant allele fraction of ≤35%, which is a significant predictor of variant pathogenicity (PP4_Moderate, PMID: 34906512, Internal lab contributor: Invitae). This variant resides within a codon (NM_00546.4: 175, 245, 248, 249, 273, 282) of TP53 that is defined as a mutational hotspot by the ClinGen TP53 VCEP (PM1; PMID: 8023157). Another missense variant(c.524G>A, p.Arg175His) (ClinVar Variation ID: 12374), in the same codon has been classified as pathogenic for Li-Fraumeni syndrome by the ClinGen TP53 VCEP’s specifications. (PM5). In summary, this variant meets the criteria to be classified as Likely Pathogenic for Li Fraumeni syndrome based on the ACMG/AMP criteria applied, as specified by the ClinGen TP53 VCEP: PS4_Moderate, PM2_Supporting, BS3_Supporting, PP3_Moderate, PP4, PM1, PM5 (Bayesian Points: 9; VCEP specifications version 2.0; 9/6/2024)   \n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      The c.305C>G variant in BRCA1 is a missense variant predicted to cause substitution of Alanine by Glycine at amino acid 102 (p.Ala102Gly). This variant is absent from gnomAD v2.1 (exomes only, non-cancer subset, read depth ≥25) and gnomAD v3.1 (non-cancer subset, read depth ≥25) (PM2_Supporting met). This missense variant is located outside of a key functional domain and was not predicted to alter mRNA splicing using the SpliceAI predictor (score 0.00, score threshold <0.1) (BP1_Strong met). Reported by one calibrated study to exhibit protein function similar to benign control variants (PMID: 30219179) (BS3 met). Multifactorial likelihood ratio analysis using clinically calibrated data produced a combined LR for this variant of 0.015 (based on Co-occurrence LR=1.177; Family History LR=0.0127), below the threshold for strong benign evidence (LR <0.05) (BP5_Strong met; PMID: 31131967, 31853058). In summary, this variant meets the criteria to be classified as a Benign variant for BRCA1-related cancer predisposition based on the ACMG/AMP criteria applied as specified by the ENIGMA BRCA1/2 VCEP (PM2_Supporting, BP1_Strong, BS3, BP5_Strong).   \n",
       "\n",
       "                                                                            PubMed Articles  \\\n",
       "72  1562462, 25584008, 27210295, 10435620, 21343334, 25584008, 30224644, 12826609, 29979965   \n",
       "68                                                                                      NaN   \n",
       "0                                                                                  30219179   \n",
       "\n",
       "                   Expert Panel Guideline Approval Date Published Date  \\\n",
       "72                    TP53 VCEP       NaN    2020-08-11     2020-08-14   \n",
       "68                    TP53 VCEP       NaN    2024-09-06     2024-09-06   \n",
       "0   ENIGMA BRCA1 and BRCA2 VCEP       NaN    2024-06-12     2024-06-11   \n",
       "\n",
       "    Retracted  \\\n",
       "72      False   \n",
       "68      False   \n",
       "0       False   \n",
       "\n",
       "                                                                  Evidence Repo Link  \\\n",
       "72  https://erepo.genome.network/evrepo/ui/classification/CA000424/MONDO:0018875/009   \n",
       "68  https://erepo.genome.network/evrepo/ui/classification/CA000252/MONDO:0018875/009   \n",
       "0   https://erepo.genome.network/evrepo/ui/classification/CA002009/MONDO:0011450/092   \n",
       "\n",
       "                                    Uuid GeneSymbol  \\\n",
       "72  ca16e21b-853c-433e-88e5-601b2de6c525       TP53   \n",
       "68  6aa7e5db-edd6-4608-a046-e82aa9ea21df       TP53   \n",
       "0   de5081dd-8e3e-444d-9826-44632cbee0f3      BRCA1   \n",
       "\n",
       "                                                                    metEvidence  \\\n",
       "72                                        [PP3, PS4_Supporting, BS3_Supporting]   \n",
       "68  [PM1, PM5, PP4, PM2_Supporting, BS3_Supporting, PS4_Moderate, PP3_Moderate]   \n",
       "0                                 [PM2_Supporting, BP1_Strong, BS3, BP5_Strong]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Expressions  \\\n",
       "72                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 (NM_000546.5:c.800G>A, NC_000017.11:g.7673820C>T, CM000679.2:g.7673820C>T, NC_000017.10:g.7577138C>T, CM000679.1:g.7577138C>T, NC_000017.9:g.7517863C>T, NG_017013.2:g.18731G>A, LRG_321:g.18731G>A, LRG_321t1:c.800G>A, NM_001126112.2:c.800G>A, LRG_321t2:c.800G>A, NM_001126113.2:c.800G>A, LRG_321t4:c.800G>A, NM_001126114.2:c.800G>A, LRG_321t3:c.800G>A, NM_001126115.1:c.404G>A, LRG_321t5:c.404G>A, NM_001126116.1:c.404G>A, LRG_321t6:c.404G>A, NM_001126117.1:c.404G>A, LRG_321t7:c.404G>A, NM_001126118.1:c.683G>A, LRG_321t8:c.683G>A, NM_001276695.1:c.683G>A, NM_001276696.1:c.683G>A, NM_001276697.1:c.323G>A, NM_001276698.1:c.323G>A, NM_001276699.1:c.323G>A, NM_001276760.1:c.683G>A, NM_001276761.1:c.683G>A, NM_001276695.2:c.683G>A, NM_001276696.2:c.683G>A, NM_001276697.2:c.323G>A, NM_001276698.2:c.323G>A, NM_001276699.2:c.323G>A, NM_001276760.2:c.683G>A, NM_001276761.2:c.683G>A, ENST00000269305.8:c.800G>A, ENST00000359597.8:n.800G>A, ENST00000413465.6:n.782+361G>A, ENST00000420246.6:c.800G>A, ENST00000445888.6:c.800G>A, ENST00000455263.6:c.800G>A, ENST00000504290.5:c.404G>A, ENST00000504937.5:c.404G>A, ENST00000509690.5:c.404G>A, ENST00000510385.5:c.404G>A, ENST00000610292.4:c.683G>A, ENST00000610538.4:c.683G>A, ENST00000610623.4:c.323G>A, ENST00000615910.4:n.767G>A, ENST00000617185.4:c.800G>A, ENST00000618944.4:c.323G>A, ENST00000619186.4:c.323G>A, ENST00000619485.4:c.683G>A, ENST00000620739.4:c.683G>A, ENST00000622645.4:c.683G>A, ENST00000635293.1:c.683G>A, NM_000546.5(TP53):c.800G>A (p.Arg267Gln))   \n",
       "68  (NM_000546.6:c.524G>T, NC_000017.11:g.7675088C>A, CM000679.2:g.7675088C>A, NC_000017.10:g.7578406C>A, CM000679.1:g.7578406C>A, NC_000017.9:g.7519131C>A, NG_017013.2:g.17463G>T, LRG_321:g.17463G>T, ENST00000503591.2:c.524G>T, ENST00000508793.6:c.524G>T, ENST00000509690.6:c.128G>T, ENST00000514944.6:c.245G>T, ENST00000604348.6:c.503G>T, ENST00000269305.9:c.524G>T, ENST00000269305.8:c.524G>T, ENST00000359597.8:c.524G>T, ENST00000413465.6:c.524G>T, ENST00000420246.6:c.524G>T, ENST00000445888.6:c.524G>T, ENST00000455263.6:c.524G>T, ENST00000504290.5:c.128G>T, ENST00000504937.5:c.128G>T, ENST00000505014.5:n.780G>T, ENST00000509690.5:c.128G>T, ENST00000510385.5:c.128G>T, ENST00000514944.5:c.245G>T, ENST00000574684.1:n.32G>T, ENST00000610292.4:c.407G>T, ENST00000610538.4:c.407G>T, ENST00000610623.4:c.47G>T, ENST00000615910.4:c.491G>T, ENST00000617185.4:c.524G>T, ENST00000618944.4:c.47G>T, ENST00000619186.4:c.47G>T, ENST00000619485.4:c.407G>T, ENST00000620739.4:c.407G>T, ENST00000622645.4:c.407G>T, ENST00000635293.1:c.407G>T, NM_000546.5:c.524G>T, LRG_321t1:c.524G>T, NM_001126112.2:c.524G>T, LRG_321t2:c.524G>T, NM_001126113.2:c.524G>T, LRG_321t4:c.524G>T, NM_001126114.2:c.524G>T, LRG_321t3:c.524G>T, NM_001126115.1:c.128G>T, LRG_321t5:c.128G>T, NM_001126116.1:c.128G>T, LRG_321t6:c.128G>T, NM_001126117.1:c.128G>T, LRG_321t7:c.128G>T, NM_001126118.1:c.407G>T, LRG_321t8:c.407G>T, NM_001276695.1:c.407G>T, NM_001276696.1:c.407G>T, NM_001276697.1:c.47G>T, NM_001276698.1:c.47G>T, NM_001276699.1:c.47G>T, NM_001276760.1:c.407G>T, NM_001276761.1:c.407G>T, NM_001276695.2:c.407G>T, NM_001276696.2:c.407G>T, NM_001276697.2:c.47G>T, NM_001276698.2:c.47G>T, NM_001276699.2:c.47G>T, NM_001276760.2:c.407G>T, NM_001276761.2:c.407G>T, NM_001126112.3:c.524G>T, NM_001126113.3:c.524G>T, NM_001126114.3:c.524G>T, NM_001126115.2:c.128G>T, NM_001126116.2:c.128G>T, NM_001126117.2:c.128G>T, NM_001126118.2:c.407G>T, NM_001276695.3:c.407G>T, NM_001276696.3:c.407G>T, NM_001276697.3:c.47G>T, NM_001276698.3:c.47G>T, NM_001276699.3:c.47G>T, NM_001276760.3:c.407G>T, NM_001276761.3:c.407G>T, NM_000546.6(TP53):c.524G>T (p.Arg175Leu))   \n",
       "0                                                                                                                                                                                      (NM_007294.4:c.305C>G, NC_000017.11:g.43104258G>C, CM000679.2:g.43104258G>C, NC_000017.10:g.41256275G>C, CM000679.1:g.41256275G>C, NC_000017.9:g.38509801G>C, NG_005905.2:g.113726C>G, LRG_292:g.113726C>G, ENST00000354071.8:n.369C>G, ENST00000461574.2:c.305C>G, ENST00000470026.6:c.305C>G, ENST00000473961.6:c.305C>G, ENST00000476777.6:c.305C>G, ENST00000477152.6:c.227C>G, ENST00000478531.6:c.305C>G, ENST00000489037.2:c.227C>G, ENST00000493919.6:c.164C>G, ENST00000494123.6:c.305C>G, ENST00000497488.2:c.-218-9398C>G, ENST00000618469.2:c.305C>G, ENST00000634433.2:c.305C>G, ENST00000644379.2:c.305C>G, ENST00000644555.2:c.164C>G, ENST00000652672.2:c.164C>G, ENST00000484087.6:c.305C>G, ENST00000700083.1:n.1276C>G, ENST00000700182.1:c.227C>G, ENST00000700183.1:c.*219C>G, ENST00000700184.1:n.548C>G, ENST00000357654.9:c.305C>G, ENST00000471181.7:c.305C>G, ENST00000642945.1:c.*179C>G, ENST00000644555.1:c.164C>G, ENST00000652672.1:c.164C>G, ENST00000352993.7:c.305C>G, ENST00000354071.7:c.305C>G, ENST00000357654.7:c.305C>G, ENST00000461221.5:c.*91C>G, ENST00000461798.5:c.*91C>G, ENST00000468300.5:c.305C>G, ENST00000470026.5:c.305C>G, ENST00000471181.6:c.305C>G, ENST00000473961.5:c.28C>G, ENST00000476777.5:c.305C>G, ENST00000477152.5:c.227C>G, ENST00000478531.5:c.305C>G, ENST00000484087.5:c.53C>G, ENST00000487825.5:c.53C>G, ENST00000489037.1:c.227C>G, ENST00000491747.6:c.305C>G, ENST00000492859.5:c.*241C>G, ENST00000493795.5:c.164C>G, ENST00000493919.5:c.164C>G, ENST00000494123.5:c.305C>G, ENST00000497488.1:c.-218-9398C>G, ENST00000586385.5:c.4+20924C>G, ENST00000591534.5:c.-44+21013C>G, ENST00000591849.5:c.-99+21013C>G, ENST00000634433.1:c.305C>G, NM_007294.3:c.305C>G, LRG_292t1:c.305C>G, NM_007297.3:c.164C>G, NM_007298.3:c.305C>G, NM_007299.3:c.305C>G, NM_007300.3:c.305C>G, NR_027676.1:n.444C>G, NM_007297.4:c.164C>G, NM_007299.4:c.305C>G, NM_007300.4:c.305C>G, NR_027676.2:n.485C>G, NM_007294.4(BRCA1):c.305C>G (p.Ala102Gly))   \n",
       "\n",
       "       hgvs_pro    scoreset_source  strength  strengthAbs  \n",
       "72  p.Arg267Gln          Kato_TP53         6            6  \n",
       "68  p.Arg175Leu       Giacomelli_1         6            6  \n",
       "0   p.Ala102Gly  Starita_BRCA1_HDR         0            0  "
      ]
     },
     "execution_count": 324,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfMax.loc[(dfMax.loc[:,'Applied Evidence Codes (Met)'].str.contains(\"BS3\")) & \\\n",
    "     (dfMax.loc[:,'strength'] >= 0) & \\\n",
    "          (dfMax.loc[:,'GeneSymbol'] != \"PTEN\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0.52617947, 0.54530177, 0.57726334, 0.621791  , 0.70374374,\n",
       "              nan,        nan,        nan])"
      ]
     },
     "execution_count": 320,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thresholdsP['Giacomelli_2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 0.07543938,  0.0481218 ,  0.00441367, -0.06388029,         nan,\n",
       "               nan,         nan,         nan])"
      ]
     },
     "execution_count": 321,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thresholdsB['Giacomelli_2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>accession</th>\n",
       "      <th>hgvs_nt</th>\n",
       "      <th>hgvs_splice</th>\n",
       "      <th>score</th>\n",
       "      <th>author_transcript</th>\n",
       "      <th>is_synonymous</th>\n",
       "      <th>scores</th>\n",
       "      <th>num_p_lp</th>\n",
       "      <th>num_b_lb</th>\n",
       "      <th>num_conflicting</th>\n",
       "      <th>num_VUS</th>\n",
       "      <th>clinvar_alleleIDs</th>\n",
       "      <th>clinvar_records</th>\n",
       "      <th>clinvar_spliceAI_max</th>\n",
       "      <th>AF</th>\n",
       "      <th>gnomAD_spliceAI_score</th>\n",
       "      <th>gnomAD_variants_VCF_INFO</th>\n",
       "      <th>labels</th>\n",
       "      <th>author_labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6733</th>\n",
       "      <td>6733</td>\n",
       "      <td>p.Ala69Gly</td>\n",
       "      <td>urn:mavedb:00000068-b-1#6750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.012334</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>False</td>\n",
       "      <td>[1.0123342207]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>[VUS, non-synonymous]</td>\n",
       "      <td>N/A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      index    hgvs_pro                     accession  hgvs_nt  hgvs_splice  \\\n",
       "6733   6733  p.Ala69Gly  urn:mavedb:00000068-b-1#6750      NaN          NaN   \n",
       "\n",
       "         score author_transcript  is_synonymous          scores  num_p_lp  \\\n",
       "6733  1.012334         NM_000546          False  [1.0123342207]       0.0   \n",
       "\n",
       "      num_b_lb  num_conflicting  num_VUS  \\\n",
       "6733       0.0              0.0     14.0   \n",
       "\n",
       "                                                                                      clinvar_alleleIDs  \\\n",
       "6733  230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112|230112   \n",
       "\n",
       "      clinvar_records  clinvar_spliceAI_max  AF  gnomAD_spliceAI_score  \\\n",
       "6733             14.0                  0.13 NaN                    NaN   \n",
       "\n",
       "     gnomAD_variants_VCF_INFO                 labels author_labels  \n",
       "6733                     None  [VUS, non-synonymous]           N/A  "
      ]
     },
     "execution_count": 318,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scoresets['Giacomelli_2'][scoresets['Giacomelli_2'].hgvs_pro == \"p.Ala69Gly\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>ProtDescription</th>\n",
       "      <th>AAchange</th>\n",
       "      <th>Codon_Number</th>\n",
       "      <th>WAF1nWT</th>\n",
       "      <th>MDM2nWT</th>\n",
       "      <th>BAXnWT</th>\n",
       "      <th>h1433snWT</th>\n",
       "      <th>AIP1nWT</th>\n",
       "      <th>GADD45nWT</th>\n",
       "      <th>NOXAnWT</th>\n",
       "      <th>P53R2nWT</th>\n",
       "      <th>WAF1nWT_Saos2</th>\n",
       "      <th>MDM2nWT_Saos2</th>\n",
       "      <th>BAXnWT_Saos2</th>\n",
       "      <th>h1433snWT_Saos2</th>\n",
       "      <th>AIP1nWT_Saos2</th>\n",
       "      <th>PUMAnWT_Saos2</th>\n",
       "      <th>SubG1nWT_Saos2</th>\n",
       "      <th>Oligomerisation_yeast</th>\n",
       "      <th>author_transcript</th>\n",
       "      <th>is_synonymous</th>\n",
       "      <th>scores</th>\n",
       "      <th>num_p_lp</th>\n",
       "      <th>num_b_lb</th>\n",
       "      <th>num_conflicting</th>\n",
       "      <th>num_VUS</th>\n",
       "      <th>clinvar_alleleIDs</th>\n",
       "      <th>clinvar_records</th>\n",
       "      <th>clinvar_spliceAI_max</th>\n",
       "      <th>AF</th>\n",
       "      <th>gnomAD_spliceAI_score</th>\n",
       "      <th>gnomAD_variants_VCF_INFO</th>\n",
       "      <th>labels</th>\n",
       "      <th>author_labels</th>\n",
       "      <th>scores_pre_normalize</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1734</th>\n",
       "      <td>1734</td>\n",
       "      <td>p.Arg267Gln</td>\n",
       "      <td>p.R267Q</td>\n",
       "      <td>R267Q</td>\n",
       "      <td>267</td>\n",
       "      <td>16.1</td>\n",
       "      <td>23.6</td>\n",
       "      <td>18.3</td>\n",
       "      <td>16.9</td>\n",
       "      <td>19.3</td>\n",
       "      <td>78.3</td>\n",
       "      <td>61.8</td>\n",
       "      <td>123.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>False</td>\n",
       "      <td>[-2.2402917724]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>458574|458574|458574|458574|458574|458574|458574|127823|127823|127823|127823|127823|127823|127823|127823|127823|127823|127823|127823</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.08</td>\n",
       "      <td>17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T</td>\n",
       "      <td>[gnomAD, VUS, conflicting, non-synonymous]</td>\n",
       "      <td>Intermediate</td>\n",
       "      <td>[21.45]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      index     hgvs_pro ProtDescription AAchange  Codon_Number  WAF1nWT  \\\n",
       "1734   1734  p.Arg267Gln         p.R267Q    R267Q           267     16.1   \n",
       "\n",
       "      MDM2nWT  BAXnWT  h1433snWT  AIP1nWT  GADD45nWT  NOXAnWT  P53R2nWT  \\\n",
       "1734     23.6    18.3       16.9     19.3       78.3     61.8     123.0   \n",
       "\n",
       "      WAF1nWT_Saos2  MDM2nWT_Saos2  BAXnWT_Saos2  h1433snWT_Saos2  \\\n",
       "1734            NaN            NaN           NaN              NaN   \n",
       "\n",
       "      AIP1nWT_Saos2  PUMAnWT_Saos2  SubG1nWT_Saos2 Oligomerisation_yeast  \\\n",
       "1734            NaN            NaN             NaN                  None   \n",
       "\n",
       "     author_transcript  is_synonymous           scores  num_p_lp  num_b_lb  \\\n",
       "1734         NM_000546          False  [-2.2402917724]       0.0       0.0   \n",
       "\n",
       "      num_conflicting  num_VUS  \\\n",
       "1734              7.0     12.0   \n",
       "\n",
       "                                                                                                                         clinvar_alleleIDs  \\\n",
       "1734  458574|458574|458574|458574|458574|458574|458574|127823|127823|127823|127823|127823|127823|127823|127823|127823|127823|127823|127823   \n",
       "\n",
       "      clinvar_records  clinvar_spliceAI_max        AF  gnomAD_spliceAI_score  \\\n",
       "1734             19.0                  0.08  0.000004                   0.08   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                gnomAD_variants_VCF_INFO  \\\n",
       "1734  17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673703:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T|17:7673820:C:T   \n",
       "\n",
       "                                          labels author_labels  \\\n",
       "1734  [gnomAD, VUS, conflicting, non-synonymous]  Intermediate   \n",
       "\n",
       "     scores_pre_normalize  \n",
       "1734              [21.45]  "
      ]
     },
     "execution_count": 327,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scoresets['Kato_TP53'][scoresets['Kato_TP53'].hgvs_pro == \"p.Arg267Gln\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "metadata": {},
   "outputs": [],
   "source": [
    "kato_ss = scoresets[\"Kato_TP53\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['index', 'hgvs_pro', 'ProtDescription', 'AAchange', 'Codon_Number',\n",
       "       'WAF1nWT', 'MDM2nWT', 'BAXnWT', 'h1433snWT', 'AIP1nWT', 'GADD45nWT',\n",
       "       'NOXAnWT', 'P53R2nWT', 'WAF1nWT_Saos2', 'MDM2nWT_Saos2', 'BAXnWT_Saos2',\n",
       "       'h1433snWT_Saos2', 'AIP1nWT_Saos2', 'PUMAnWT_Saos2', 'SubG1nWT_Saos2',\n",
       "       'Oligomerisation_yeast', 'author_transcript', 'is_synonymous', 'scores',\n",
       "       'num_p_lp', 'num_b_lb', 'num_conflicting', 'num_VUS',\n",
       "       'clinvar_alleleIDs', 'clinvar_records', 'clinvar_spliceAI_max', 'AF',\n",
       "       'gnomAD_spliceAI_score', 'gnomAD_variants_VCF_INFO', 'labels',\n",
       "       'author_labels', 'scores_pre_normalize'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 334,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kato_ss.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {},
   "outputs": [],
   "source": [
    "katoblbscores = np.concatenate(kato_ss.loc[kato_ss.num_b_lb > 0,'scores_pre_normalize'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(np.float64(92.41856060606061), np.float64(31.678266857736443))"
      ]
     },
     "execution_count": 336,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "katoblbscores.mean(), katoblbscores.std()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([27.19976659, 26.91420474, 26.55725242, 25.98612871, 25.05805268,\n",
       "       23.05911971,         nan,         nan])"
      ]
     },
     "execution_count": 338,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thresholdsP['Kato_TP53'] * katoblbscores.std() + katoblbscores.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([47.61743915, 48.04578193, 48.6882961 , 49.54498166, 50.90140047,\n",
       "       54.0175942 ,         nan,         nan])"
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thresholdsB['Kato_TP53'] * katoblbscores.std() + katoblbscores.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "metadata": {},
   "outputs": [],
   "source": [
    "vampseq = scoresets['Matreyek_PTEN_VampSeq']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0.49045362, 0.48483205, 0.477524  , 0.46796733, 0.45335123,\n",
       "       0.42749199, 0.37464919,        nan])"
      ]
     },
     "execution_count": 345,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thresholdsP['Matreyek_PTEN_VampSeq']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(671, 45)"
      ]
     },
     "execution_count": 348,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vampseq[(vampseq.author_labels == \"Intermediate\") & (vampseq.scores.apply(lambda s: s[0] > thresholdsP['Matreyek_PTEN_VampSeq'][0]))].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1013, 45)"
      ]
     },
     "execution_count": 350,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vampseq[vampseq.author_labels == \"Intermediate\"].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>accession</th>\n",
       "      <th>hgvs_nt</th>\n",
       "      <th>hgvs_splice</th>\n",
       "      <th>score</th>\n",
       "      <th>sd</th>\n",
       "      <th>expts</th>\n",
       "      <th>se</th>\n",
       "      <th>lower_ci</th>\n",
       "      <th>upper_ci</th>\n",
       "      <th>score1</th>\n",
       "      <th>score2</th>\n",
       "      <th>score3</th>\n",
       "      <th>score4</th>\n",
       "      <th>score5</th>\n",
       "      <th>score6</th>\n",
       "      <th>score7</th>\n",
       "      <th>score8</th>\n",
       "      <th>median_w_ave</th>\n",
       "      <th>exp1_w_ave</th>\n",
       "      <th>exp2_w_ave</th>\n",
       "      <th>exp3_w_ave</th>\n",
       "      <th>exp4_w_ave</th>\n",
       "      <th>exp5_w_ave</th>\n",
       "      <th>exp6_w_ave</th>\n",
       "      <th>exp7_w_ave</th>\n",
       "      <th>exp8_w_ave</th>\n",
       "      <th>snv</th>\n",
       "      <th>abundance_class</th>\n",
       "      <th>author_transcript</th>\n",
       "      <th>is_synonymous</th>\n",
       "      <th>scores</th>\n",
       "      <th>num_p_lp</th>\n",
       "      <th>num_b_lb</th>\n",
       "      <th>num_conflicting</th>\n",
       "      <th>num_VUS</th>\n",
       "      <th>clinvar_alleleIDs</th>\n",
       "      <th>clinvar_records</th>\n",
       "      <th>clinvar_spliceAI_max</th>\n",
       "      <th>AF</th>\n",
       "      <th>gnomAD_spliceAI_score</th>\n",
       "      <th>gnomAD_variants_VCF_INFO</th>\n",
       "      <th>labels</th>\n",
       "      <th>author_labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [index, hgvs_pro, accession, hgvs_nt, hgvs_splice, score, sd, expts, se, lower_ci, upper_ci, score1, score2, score3, score4, score5, score6, score7, score8, median_w_ave, exp1_w_ave, exp2_w_ave, exp3_w_ave, exp4_w_ave, exp5_w_ave, exp6_w_ave, exp7_w_ave, exp8_w_ave, snv, abundance_class, author_transcript, is_synonymous, scores, num_p_lp, num_b_lb, num_conflicting, num_VUS, clinvar_alleleIDs, clinvar_records, clinvar_spliceAI_max, AF, gnomAD_spliceAI_score, gnomAD_variants_VCF_INFO, labels, author_labels]\n",
       "Index: []"
      ]
     },
     "execution_count": 352,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vampseq[vampseq.author_labels == \"Functionally Abnormal\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "author_labels\n",
       "Functionally_Normal      2478\n",
       "Intermediate             1013\n",
       "Functionally_Abnormal     918\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 353,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vampseq.author_labels.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mave",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
